# Validation Report - data.table@1.16.4


# Context

This report is fully automated and builds on rhub/ref-image image.
Documents the installation of this package on an open source R
environment, focusing on:

- Installation environment description
- Testing coverage

It is limited to assess whether unit tests and documentation are present
and can execute without error. An assessment would be required that the
tests and documentation are meaningful.

# Package data.table

## Metric based risk assessment

The following metrics are derived from the `riskmetric` R package.

|                      |                                          |
|:---------------------|:-----------------------------------------|
| has_news             | 1                                        |
| news_current         | TRUE                                     |
| exported_namespace   | 120                                      |
| has_vignettes        | 25                                       |
| export_help          | 120                                      |
| has_website          | TRUE                                     |
| has_maintainer       | Tyson Barrett <t.barrett88@gmail.com>    |
| bugs_status          | subscript out of bounds                  |
| has_source_control   | https://github.com/Rdatatable/data.table |
| has_bug_reports_url  | 1                                        |
| downloads_1yr        | 9897084                                  |
| reverse_dependencies | 1958                                     |
| has_examples         | 1                                        |
| dependencies         | 1                                        |
| license              | MPL-2.0 \| file LICENSE                  |

**Package general assessment:** Coverage, check results, size, download
the last year, reverse dependencies and number of dependencies.

## Dependencies

Overall the package has these dependencies:

|     | package | type    |
|:----|:--------|:--------|
| 1   | methods | Imports |

**Package dependencies**

## Reverse dependencies

Overall the package has 1958 reverse dependencies:

Ac3net, accessibility, acdcR, Achilles, acoRn, actel, ActivePathways,
ActivityIndex, actuaRE, ADAMgui, adductomicsR, AdhereR, AdhereRViz,
ADImpute, adjustedCurves, AeRobiology, AF, AFM, africamonitor, agcounts,
AggregateR, agriutilities, ahw, AICcPermanova, airship, akc, akiFlagger,
aLFQ, alpaca, AlphaBeta, amadeus, amapGeocode, AMAPVox, amerifluxr,
AMIM, amplican, AMR, amt, Anaconda, analyzer, animint2,
antaresEditObject, antaresProcessing, antaresRead, antaresViz,
AnxietySleep, aopdata, aorsf, AOV1R, APCI, APIS, aplotExtra, AQEval,
aqp, arcgisgeocode, arcgisutils, arctools, arf, artMS, asciiSetupReader,
ascotraceR, aslib, ASpli, ASRgenomics, assertable, atime, AtmChile,
atSNP, attrib, AUCell, aum, autoCovariateSelection, autonewsmd,
autonomics, AutoPlots, autostats, autostsm, aweSOM, bagged.outliertrees,
bakR, bambu, BANDITS, Banksy, baseballr, basecallQC, BatchExperiments,
BatchJobs, BatchQC, batchtools, BayesianMCPMod, bayesLife, BayesLN,
bayesMeanScale, bayesMig, bayesPop, BayesSUR, bayesTFR, bbk, BBmisc,
bbmix, bbotk, bcputility, bdots, bea.R, BEARscc, BEclear, BEDMatrix,
bedr, BeeGUTS, behavr, benford.analysis, bfw, BGData, BgeeCall, BgeeDB,
BGmisc, BiasCorrector, biblionetwork, bife, BIGL, bigreadr, bigsnpr,
bigstatsr, bikeshare14, binancer, binomialRF, binsegRcpp, biobroom,
BiocParallel, biomartr, BIOMASS, BioNAR, biopixR, bioregion, birankr,
biscuiteer, bkmrhat, BKTR, BLA, blorr, BMisc, bnbc, BootstrapQTL,
BPRMeth, bRacatus, brainGraph, branchpointer, brandwatchR, bratteli,
bread, BREADR, bridger2, BRINDA, brmsmargins, brpop, bruceR, bsitar,
bsseq, BTM, BTSPAS, BTYDplus, bupaR, BuyseTest, bysykkel, c3, CAGEr,
CALF, calmr, CamelUp, camtrapR, CaMutQC, CareDensity, caretEnsemble,
cartogramR, casebase, CAST, CATALYST, catcont, categoryEncodings,
CatEncoders, CatsCradle, causalCmprsk, CausalGPS, causalOT, cbl, CBRT,
ccdR, ccmap, ccostr, cdid, cdparcoord, ceas, celda, CellBarcode,
cellbaseR, CelliD, cellKey, CellNOptR, CEMiTool, Census2016, cercospoRa,
Certara.NLME8, Certara.R, Certara.RsNLME, Certara.RsNLME.ModelBuilder,
Certara.VPCResults, Certara.Xpose.NLME, cffdrs, cfr, cglm,
changepoint.influence, ChannelAttributionApp, cheapr, checkmate,
chem.databases, ChemoSpecUtils, Chicago, chicane, chimeraviz,
ChineseNames, ChIPexoQual, ChIPpeakAnno, chronicle, cicero, circhelp,
citationchaser, citecorp, ciw, clam, classifierplots, cleanTS,
ClickHouseHTTP, clickstream, climate, clinDataReview, clinspacy,
clinUtils, CluMP, ClustAssess, Clustering, clustifyr, clustra, CLVTools,
cmapR, cmsaf, CNVRanger, CNVScope, COCOA, coder, CoDiNA, coffee,
cohorts, coinmarketcapr, collapse, coloc, colocPropTest, colorDF,
colorednoise, Colossus, comorbidity, CompareCausalNetworks, compareDF,
COMPASS, ComplexUpset, ComPrAn, compSPOT, CONFESS, CoNI, conleyreg,
consensusDE, consortr, ConsReg, constellation, constructive, container,
CoOL, cooltools, CooRTweet, CopernicusDEM, cophescan, coRdon, CoreGx,
CornerstoneR, corporaexplorer, corpustools, corrcoverage,
correspondenceTables, CoSMoS, COTAN, counterfactuals, countries,
coveffectsplot, COVID19, covid19br, covid19india, covidnor, cppRouting,
CRE, creditmodel, crew, crfsuite, CRISPRseek, CRMetrics, CropScapeR,
crosshap, crossmeta, crumble, cryptoQuotes, csa, csalert, csdata,
csmaps, CSMES, cstidy, cstime, CSTools, csutil, csv, csvy, ctbi, ctmm,
cTRAP, ctsem, ctxR, cubar, curtailment, cusum, cusumcharter, cvAUC,
cvms, CWT, cxhull, cyclestreets, cylcop, cytoKernel, CytoML,
D3partitionR, daiquiri, daiR, damr, dang, dartR.base, dartR.popgen,
dartR.sim, dartR.spatial, dat, data.table.threads, data360r,
DataCombine, dataCompare, dataCompareR, datadictionary, DataExplorer,
dataframeexplorer, datamods, dataMojo, datapasta, dataprep,
dataPreparation, dataRetrieval, datarobot, datasetjson, DataSpaceR,
DATAstudio, dataverifyr, datawizard, datazoom.amazonia, dateutils,
dbi.table, dblr, dbnR, dbWebForms, ddtlcm, DeclareDesign,
decompTumor2Sig, deconvR, dedupewider, deepdive, deeptrafo, DEFormats,
DegNorm, delayed, deltaccd, DemoKin, denvax, DeProViR, DEScan2,
DescTools, designit, DESpace, detectRUNS, DEWSeq, dfmeta, DGM, DIAlignR,
diathor, diceplot, did2s, Diderot, DiDforBigData, didimputation,
DIDmultiplegtDYN, difconet, DifferentialRegulation, DiscoRhythm, DiSCos,
discSurv, DisImpact, disk.frame, distillML, DIZtools, DIZutils, DMCFB,
dMrs, do, DominoEffect, doremi, DoubleML, DQAgui, DQAstats, dQTG.seq,
drake, DramaAnalysis, dreamlet, drgee, dropR, droptest, dsos, DTAT,
dtplyr, DTSg, dtts, DTwrappers, DTwrappers2, dumbbell, Durga, dvmisc,
dynamite, DYNATE, dynsurv, DysPIA, E4tools, Eagle, easybio, easycsv,
easyr, easysurv, eatGADS, eatTools, EBPRS, eclust, ecocomDP, ecoregime,
ecospace, ecotox, edeaR, editbl, edl, EdSurvey, eeptools, efdm,
effClust, EGM, EHR, EIAapi, EIX, eixport, electionsBR, ELMER, eltr,
EmiR, EmissV, enderecobr, endogeneity, endoSwitch, enmSdmX, ensr,
envalysis, enviGCMS, eoR, epanetReader, Epi, epialleleR, epiCleanr,
EpiCompare, epidm, EpiForsk, epigraHMM, EpiMix, EpiNow2, epiomics,
epistasisGA, epiworldR, eplusr, epm, epwshiftr, equateMultiple, eSIR,
estadistica, EthSEQ, etm, etwfe, eurocordexr, eurodata, europeanaR,
eurostat, evapoRe, eventglm, eventstudyr, evoper, EWCE, exiftoolr,
explainer, exploratory, expowo, expss, exvatools, eyeRead, eyeTrackR,
ezr, fabricatr, facmodCS, facmodTS, FACT, factor256, factR, faers,
faersquarterlydata, FairMclus, familiar, FAOSTAT, farrell, fastai,
fastdid, fastDummies, fasterRaster, fastLink, fastreeR, FastRet,
fastRhockey, fastText, fastverse, fastVoteR, fbRads, fcfdr, FCO, fdq,
FeatureImpCluster, fedmatch, fedstatAPIr, fedup, felp, fgdr, Fgmutils,
fgsea, fhircrackr, FIESTA, FIESTAutils, filecacher, finbif, finch,
finnsurveytext, fishmethods, fishpond, fitbitViz, FitHiC, fixest,
flextable, flightsbr, flippant, FLOPART, florabr, flowGraph, flowml,
flowTraceR, flowWorkspace, flps, Fluidigm, FMAT, fmeffects, FOCI,
forceplate, forecastML, forestError, forImage, formatdown, formulaic,
FORTLS, fossilbrush, FourWayHMM, fplot, fplyr, FRASER, fruclimadapt,
fsbrain, fst, FuncDiv, funchir, future.tests, FuzzyDBScan, fwlplot, fy,
GALLO, GARCOM, gasfluxes, gatom, GaussSuppression, gbp, gcite, gde,
gDRcore, gDRimport, gDRutils, gdxdt, gemma.R, gems, GeneGeneInteR,
geneplast, GENESIS, GeneStructureTools, geneXtendeR, genomation,
GenomAutomorphism, GenomeInfoDb, genomic.autocorr, GenomicDataCommons,
GenomicDistributions, GenomicInteractions, GenomicScores,
GenomicTools.fileHandler, GenomicTuples, GenoScan, GenVisR, geobr,
geocausal, geocodebr, GeoMongo, GeomxTools, geoperu, GEOquery,
GeoTcgaData, GeoThinneR, germinationmetrics, getDTeval, GFE, gfer,
gfoRmula, gfoRmulaICE, ggbrain, ggcyto, ggdmc, ggetho, ggfacto,
ggfigdone, ggfixest, GGIR, GGIRread, ggkegg, ggmugs, ggperiodic, ggPMX,
ggpval, ggquickeda, ggsom, ggswissmaps, ggTimeSeries, GHap,
githubinstall, gittargets, GladiaTOX, GmicR, GMMAT, gmoTree, gnrprod,
goat, gompertztrunc, googleCloudVisionR, googleVis, GOTHiC, gpboost,
grandR, GRaNIE, grattan, grattanInflators, greatR, GREENeR, greport,
GRIN2, growthcleanr, GRShiny, GSALightning, gsDesign, gsDesign2, GSODR,
GSVA, gtfs2emis, gtfs2gps, gtfsio, gtfsrouter, gtfstools, GTFSwizard,
GUIDEseq, gunsales, gustave, gwaRs, gwasforest, GWASinspector,
GWASTools, gwid, h2o, HACSim, HaDeX, hagis, hahmmr, hal9001, haldensify,
handlr, harmony, hbamr, hBayesDM, HCD, HCR, hdd, hdpGLM, healthdb,
healthyAddress, heatwaveR, heims, hereR, hermiter, HERON, hesim,
heuristicsmineR, HicAggR, HiCBricks, HiCcompare, HiCDCPlus, HiCDOC,
hicp, highfrequency, hilldiv, himach, hisse, Hmisc, HMMcopy, hoardeR,
HodgesTools, hollr, hoopR, HospitalNetwork, hot.deck, HPLB, hR,
hsrecombi, htetree, httk, hudr, hurricaneexposure, hutils, hutilscpp,
HVT, hwig, hydroloom, hydropeak, hyfo, hypervolume, HYPEtools, i2extras,
IBRtools, iCAMP, ICAMS, iCellR, IceSat2R, iCNV, ideanet, IDEATools,
IDmining, ie2misc, ie2miscdata, ieegio, iemisctext, ifCNVR, iGC, iimi,
imabc, imcRtools, imf.data, iml, immunarch, immuneSIM, implicitMeasures,
importinegi, imputeMulti, inbreedR, IncDTW, incidence2, ineAtlas,
inferr, InflectSSP, influential, Information, INLAspacetime, inplace,
installr, insurancerating, InteractiveComplexHeatmap, InterCellar,
interface, intervalaverage, inTrees, IOHanalyzer, ipeaplot, ips, irace,
iraceplot, irg, IrregLong, ISAnalytics, ISCA, iscoCrosswalks, isdparser,
IsoBayes, isobxr, isotracer, iSTATS, itol.toolkit, itsadug,
itscalledsoccer, ivdoctr, ivmte, ivtools, jab.adverse.reactions, jage,
JointFPM, joinXL, journalabbr, joyn, jrSiCKLSNMF, JSmediation, jsmodule,
jstable, jsTree, jubilee, JWileymisc, kalmanfilter, kangar00, kdry,
kerastuneR, kesernetwork, kgen, kgraph, kibior, kim, kimfilter,
KinSwingR, kisopenapi, KMD, kmeRtone, knfi, KnockoffScreen, koRpus,
kosis, KPC, lab2clean, LabourMarketAreas, labourR, LACE, lacunr, Lahman,
LambertW, lar, Largevars, latrend, lava, lavaSearch2, lay, lazytrade,
LCMSQA, ldamatch, ldaPrototype, ldsr, leafR, leakyIV, LearningStats,
leiden, lexicon, LexisNexisTools, lfe, lgr, lgrExtra, libbib, libr,
lidR, lightgbm, LilRhino, limexhub, limorhyde, limorhyde2, lineagespot,
LinkHD, lipidr, lisaClust, live, LKT, llama, LMMstar, LMPdata, lmtp,
loadeR, LocalControl, loci2path, logiBin, LogicForest, LOLA, LOMAR,
lookupTable, LOPART, lori, LPDynR, LSPFP, Luminescence, LWFBrook90R,
lwqs, LymphoSeq, m5, Maaslin2, Macarron, maditr, maftools, MAGAR,
magclass, MAGEE, maicChecks, MAICtools, MANOVA.RM, MantaID, manydist,
mapi, marginaleffects, margins, mariner, marked, maser, MAST, matman,
MatrixExtra, MatrixHMM, mau, maxaltall, maxent.ot, mbr, mbRes, mcauchyd,
mcboost, medrxivr, Melissa, memisc, memochange, mergen, MesKit,
metabinR, metabodecon, metabolomicsWorkbenchR, MetaComp, metaforest,
metaGE, metagene2, MetAlyzer, metamer, metasnf, meteo, methimpute,
methodical, MethReg, methrix, methylKit, MethylMix, methylPipe,
methylscaper, metR, mggd, mgi.report.reader, MIC, miceFast, micemd,
miceRanger, MicrobiotaProcess, microeco, MicroMoB, microseq, midfieldr,
miesmuschel, miLineage, milr, MIMER, mimi, mind, minfi, MinimumDistance,
minSNPs, mintyr, mipplot, MIRA, mirTarRnaSeq, miscset, missCompare,
misty, mixgb, mixIndependR, mixvlmc, mknapsack, mlbplotR,
mldr.resampling, mlexperiments, mllrnrs, mlr, mlr3, mlr3batchmark,
mlr3benchmark, mlr3cluster, mlr3db, mlr3fairness, mlr3fda, mlr3filters,
mlr3fselect, mlr3hyperband, mlr3inferr, mlr3learners, mlr3mbo, mlr3misc,
mlr3oml, mlr3pipelines, mlr3resampling, mlr3shiny, mlr3spatial,
mlr3spatiotempcv, mlr3summary, mlr3superlearner, mlr3torch, mlr3tuning,
mlr3tuningspaces, mlr3verse, mlr3viz, mlrintermbo, mlrMBO, mlsurvlrnrs,
mltools, mmaqshiny, mnem, mobileRNA, MOCHA, modACDC, ModCon,
ModelMetrics, modelsummary, MOFA2, MoleculeExperiment, MouseFM, MPAC,
mpactr, MplusAutomation, mpwR, mrgsolve, mrMLM, mrMLM.GUI, mrregression,
MsBackendSql, mschart, msImpute, mslp, MSmix, MSnID, MSstats,
MSstatsConvert, MSstatsLiP, MSstatsPTM, MSstatsShiny, MSstatsTMT,
mstate, mstclustering, mstrio, mstudentd, mtb, mulea, multicrispr,
multifamm, MultiFit, multiHiCcompare, multilevelcoda, multilevelTools,
multinets, multistateQTL, multiWGCNA, multvardiv, MungeSumstats, muscat,
musica, musicatk, mutationtypes, mutualinf, mvQuad, mvSUSY, mxnorm,
myClim, naaccr, NACHO, nanotime, naptanr, nc, ndjson, neatR, neatRanges,
neatStats, neo4r, NEONiso, neonOS, neonPlantEcology, neonUtilities,
nestedcv, net4pg, netCoin, netgsa, NetLogoR, netSmooth, networkR,
netZooR, neuroimaGene, nfl4th, nflfastR, nflplotR, nflreadr, nflseedR,
NFLSimulatoR, nhppp, nlmixr2est, nlmixr2extra, nlpembeds, nlpred,
NMcalc, NMdata, NMsim, nngeo, NNS, noctua, node2vec, noncompliance,
nonmem2rx, normfluodbf, nosoi, npcurePK, nRegression, nullranges,
numbat, o2geosocial, OBIC, obliqueRSF, occupationMeasurement, octad,
octopus, odbr, oddsapiR, ofpetrial, ogrdbstats, OGRE, okxAPI,
OlinkAnalyze, OmicNavigator, OMICsPCA, omicwas, ompr, OnboardClient,
OncoSimulR, One4All, onlineforecast, onsr, openCyto, opendataformat,
openEBGM, OpenML, OpenSpecy, opentripplanner, optiSel, orderanalyzer,
ORFhunteR, ORFik, orgR, origami, origin, orthogene, osmclass,
outliertree, OUTRIDER, overviewR, owdbr, owidR, paar, packageRank, padr,
pagoda2, pairkat, PakPMICS2014Ch, PakPMICS2014HH, PakPMICS2014HL,
PakPMICS2014Wm, pam, PAMpal, PAMscapes, pander, panelaggregation,
PanelMatch, panelWranglR, panstarrs, PanViz, paradox, parallelMap,
paramix, ParBayesianOptimization, parseRPDR, parSim, partCNV, partools,
parttree, pathviewr, pathwayTMB, patternator, pbox, pcadapt, PCBS,
PCMBase, PCMBaseCpp, PCRA, pcvr, pda, PDATK, pdynmc, PeakSegDisk,
PeakSegOptimal, pedalfast.data, pedbp, Pedixplorer, pedquant,
penaltyLearning, PepMapViz, pepr, pepStat, PGRdup, pgTools, phantasus,
phantasusLite, PharmacoGx, pheble, phenofit, phenomis,
PhenotypeSimulator, phers, phm, phonics, PhyloProfile, phyloseq,
PhysicalActivity, pingers, pins, pinyin, pipeflow, pivottabler, pk4adi,
pkggraph, pkgnet, PKPDsim, pksensi, PlanetNICFI, plinkQC, plmmr, plnr,
plotgardener, plotHMM, plotly, pmartR, pmcalibration, pmml, pmparser,
pmxpartab, pointblank, pointdensityP, poisFErobust, politeness, polle,
polmineR, polypharmacy, pomp, POMS, poolfstat, pooling, popEpi,
PopulateR, PortfolioAnalytics, portvine, posologyr, POSSA, potools,
POUMM, power.transform, powerbiR, PP, ppmlasso, pram, pRecipe, precrec,
predhy.GUI, prediction, PReMiuM, PreProcessRecordLinkage, preputils,
PressPurt, preventr, previsionio, pricelevels, primer, prioriactions,
prioritizr, prnsamplr, proActiv, processmapR, processpredictR, prodlim,
ProjectTemplate, prolific.api, PRONE, PROSPER, Prostar, protGear,
protti, prrd, prt, psborrow, PSF, psichomics, psidR, psmineR, psp,
psycCleaning, PsychWordVec, ptable, ptairMS, pTITAN2, PTXQC, Publish,
PureCN, puremoe, pureseqtmr, pvda, Pviz, pvLRT, qad, QCAcluster,
qckitfastq, qdapTools, qgg, qreport, qs, qs2, qte, QTL.gCIMapping,
QTL.gCIMapping.GUI, qtl2, QuartPAC, quickcheck, quickPlot, R.AlphA.Home,
r2dii.match, R2DT, r2glmm, R3CPET, r3Cseq, r5r, R62S3, R6causal, RABR,
RadEro, RadioGx, radlibs, RAINBOWR, RALSA, rAmCharts, RaMS, randnet,
randomForestExplainer, randomForestSRC, randomNames, rangeMapper,
RankAggSIgFUR, RapidoPGS, raqs, RARfreq, rassta, rasterDT, RAthena,
ratios, Ravages, RawHummus, rbacon, RBaseX, rBayesianOptimization, rbi,
rbi.helpers, rBiasCorrection, rbin, rBiopaxParser, Rblpapi, rblt,
Rborist, rbtt, Rcan, RCAS, RcextTools, rchemo, RcisTarget,
RcmdrPlugin.RiskDemo, RcmdrPlugin.WorldFlora, RCNA, rCNV,
RcppGreedySetCover, rcprd, Rcrawler, rdbnomics, rdfp, rdhs,
Rdiagnosislist, Rdice, RDML, rdracor, rdwd, ReactomeContentService4R,
ReactomeGraph4R, read.gt3x, readNSx, readtext, realTimeloads,
reappraised, rechaRge, reclin2, recometrics, recorder, RecordLinkage,
recount3, redatam, Rediscover, RefBasedMI, regport, regtools, rehh,
ReinforcementLearning, reinsureR, remiod, remotePARTS, rENA, REndo,
replacer, repmis, reproducible, Require, rerddap, resemble, reservr,
reslr, RESOLVE, respR, restatapi, ReSurv, rfaRm, rfars, rfisheries,
RFplus, rfPred, rgbif, RGMQL, rhdf5client, RIbench, RiboCrypt, RiboDiPA,
RiboProfiling, ricu, Rilostat, RImmPort, rintcal, rio, rIsing,
riskRegression, RITCH, ritis, rlas, rlist, rllama, RLumShiny, RMassBank,
rmcfs, rMIDAS, RNewsflow, rnpn, Rnvd3, roads, robCompositions, RobinCar,
robis, robotoolbox, ROCket, rODE, rollinglda, rorcid, rotor, rPackedBar,
rpivotTable, rprev, rqdatatable, rqPen, rreg, Rrepest, RSauceLabs,
rSPARCS, rsparse, rstanarm, rsurveycto, rsyntax, RTCGA, RTCGAToolbox,
rtmpt, RTN, RTNsurvival, rtop, RTransferEntropy, rtry, rtsdata,
RTTWebClient, ruimtehol, runcharter, rush, rusquant, RVA, RWDataPlyr,
rWikiPathways, RWildbook, rworkflows, RxnSim, rxode2, ryandexdirect,
rzentra, RZooRoH, s3fs, S4Vectors, saeHB.twofold, SafeVote, salesforcer,
samadb, sampcompR, samplingin, sangeranalyseR, sanityTracker, sansa,
SARC, sarp.snowprofile, sarp.snowprofile.alignment,
sarp.snowprofile.pyface, saseR, SC2API, scanMiR, scanMiRApp,
scCustomize, SCdeconR, scDiffCom, scifer, scMET, scMitoMut, SCnorm,
scoper, scopr, scorecard, ScoreEB, scoringutils, scPipe, scraEP,
scRNAseqApp, scruff, scTreeViz, sdcHierarchies, sdcLog, sdcMicro,
sdcTable, SDLfilter, SEA, seAMLess, SeaVal, seedr, segregation,
seleniumPipes, SELF, semgram, SemNeT, semtree, sendigR, sense, sensobol,
sentiment.ai, sentimentr, SenTinMixt, sentometrics, sentopics,
seqsetvis, sequoia, SeqVarTools, servosphereR, SEtools, Seurat,
sevenbridges, sevenbridges2, sevenC, sgat, SGP, SGPdata, SHAPforxgboost,
shapr, shinyChatR, shinyepico, shinyExprPortal, shinyHugePlot,
ShinyItemAnalysis, shinylogs, shinyML, ShinyQuickStarter, shinySbm,
shinyWGD, siconvr, SightabilityModel, sigminer, Signac, signatureSearch,
siie, SimBu, simcausal, simDAG, SimEngine, simfinapi, simitation, simPH,
simphony, simplePHENOTYPES, simPop, simstudy, SimSurvey, simtrial,
SingleCellSignalR, singleCellTK, sitepickR, siteymlgen, sitree, sitreeE,
sjtable2df, skimr, skm, sleepr, SLEMI, SlideCNA, slim,
SmallCountRounding, smallstuff, SmartEDA, smartsnp, smoothic, snapcount,
SNPannotator, SNPhood, snplinkage, snpsettest, socialh, socialmixr,
soilDB, SoilHyP, SoilTaxonomy, soiltestcorr, solaR2, solitude, SOMNiBUS,
SOMnmR, somspace, soundgen, SpaceMarkers, spacyr, SpaDES.core,
SpaDES.tools, sparrow, sparseFLMM, SPARSEMODr, SparseSignatures, spaths,
SpatialCPie, SpatialFeatureExperiment, spatialHeatmap,
SpatialOmicsOverlay, spatialrisk, SpATS, spatsoc, spDates, speaq,
spectr, spectralAnalysis, spellcheckr, spicyR, SpliceWiz,
splitstackshape, splutil, spNetwork, spocc, SPONGE, sport, sportyR,
sqlHelpers, SQMtools, squallms, squid, SSBtools, SSHAARP, staggered,
standartox, staRdom, statar, statcanR, statgenGWAS, statgenIBD,
statgenMPP, statgenQTLxT, Statial, Statsomat, stdReg, stdReg2, stenR,
sticky, SticsRFiles, stm, stochvolTMB, stplanr, stppSim, stringmagic,
strvalidator, subscreen, subSeq, sumFREGAT, SUMMER, suncalc, SUNGEO,
SunsVoc, superml, surveyplanning, surveyPrev, surveysd, survMisc,
susographql, SVMDO, SVN, SWATH2stats, sweater, SWMPr, symphony, synlet,
synr, table.express, tablecompare, tabledown, tableMatrix, tabulator,
tabxplor, takos, TangledFeatures, targeted, targets, taskscheduleR,
tatoo, taxize, taxonomizr, TCA, TCGAbiolinks, tcgaViz, tcpl, tcplfit2,
tdsc, TeXCheckR, text2vec, textclean, TextMiningGUI, textplot,
textpress, textrank, textrecipes, textshape, textTinyR, textTools,
TFutils, TheOpenAIR, TideCurves, TideTables, tidyDenovix, TidyDensity,
tidyfast, tidyfst, tidyft, tidygapminder, tidyplate, tidyr, tidyrules,
tidysdm, tidytable, tidyterra, tidytext, tidytransit, tidyvpc, tiledb,
timeplyr, timeseriesdb, TIN, tinytable, TitanCNA, tLOH, tmap, TMSig,
TnT, toolStability, topdowntimeratio, tosca, totalcensus, ToxicoGx,
toxSummary, TPAC, TPACData, tpm, TPP, tracee, trackdf, tradestatistics,
TrafficBDE, trainsplit, transcriptogramer, transport, traveltimeR,
treediff, treekoR, treemap, TreeMineR, treeshap, triact, TrialEmulation,
tRigon, trimmer, tripr, TrumpetPlots, tsbox, tsdb, tsdistributions,
tsgarch, tsmarch, tsmethods, TSrepr, TSstudio, tstests, tstools,
TTAinterfaceTrendAnalysis, ttgsea, Tushare, tvtools, twang, twangRDC,
twc, twl, twopartm, txshift, uavRmp, ubair, UCell, uci, udpipe, ugatsdb,
UKB.COVID19, ukbnmr, ukbtools, UKgrid, Ularcirc, Unico, unifed,
Uniquorn, uptasticsearch, utilsIPEA, vaersNDvax, vaersvax, validata,
VanillaICE, vardpoor, vein, versioning, versus, VIM, vimp, ViSEAGO,
visualpred, vivainsights, VMDecomp, VOSONDash, vosonSML, vote, votesys,
VoxR, Voyager, vtreat, vvtableau, walkboutr, wally, WARDEN, weatherOz,
WeatherSentiment, WebAnalytics, webtrackR, wehoop, WGScan, WhatsR,
where, whSample, wikitaxa, wilson, wmm, word.alignment, WorldFlora,
worrms, wpa, wpp2017, wpp2019, WQM, wrMisc, wrProteo, wTO, WVPlots,
wxgenR, wyz.code.metaTesting, wyz.code.offensiveProgramming,
wyz.code.rdoc, wyz.code.testthat, xcore, xefun, xesreadR, xgb2sql,
xgboost, XNAString, xpectr, xplorerr, yamss, YEAB, youngSwimmers, zebu,
zeitgebr, ZIM4rv, ZIprop, zoltr, zoomGroupStats

## Namespace

Overall the package has 120 exported objects. 112 are documented:

                    :=                .BY             .EACHI               .GRP 
                  TRUE               TRUE               TRUE               TRUE 
                    .I      .Last.updated                 .N              .NGRP 
                  TRUE               TRUE               TRUE               TRUE 
                   .SD          %between%             %chin%            %flike% 
                  TRUE              FALSE              FALSE              FALSE 
               %ilike%          %inrange%             %like%            %notin% 
                 FALSE              FALSE              FALSE              FALSE 
               %plike%            address          alloc.col      as.data.table 
                 FALSE               TRUE               TRUE               TRUE 
              as.IDate           as.ITime  as.xts.data.table            between 
                  TRUE               TRUE               TRUE               TRUE 
               chgroup            chmatch            chorder                 CJ 
                  TRUE               TRUE               TRUE               TRUE 
                  copy               cube         data.table              dcast 
                  TRUE               TRUE               TRUE               TRUE 
      dcast.data.table              fcase          fcoalesce        fdroplevels 
                  TRUE               TRUE               TRUE               TRUE 
               fifelse         fintersect              first         format_col 
                  TRUE               TRUE               TRUE               TRUE 
      format_list_item          foverlaps              frank             frankv 
                  TRUE               TRUE               TRUE               TRUE 
                 fread         frollapply          frollmean           frollsum 
                  TRUE               TRUE               TRUE               TRUE 
              fsetdiff          fsetequal              fsort             funion 
                  TRUE               TRUE               TRUE               TRUE 
                fwrite       getDTthreads getNumericRounding       groupingsets 
                  TRUE               TRUE               TRUE               TRUE 
                haskey               hour          IDateTime            indices 
                  TRUE               TRUE               TRUE               TRUE 
               inrange      is.data.table            isoweek                key 
                  TRUE               TRUE               TRUE               TRUE 
                 key<-               last                let               like 
                  TRUE               TRUE               TRUE               TRUE 
                  mday            measure               melt    melt.data.table 
                  TRUE               TRUE               TRUE               TRUE 
      merge.data.table             minute              month             nafill 
                  TRUE               TRUE               TRUE               TRUE 
              patterns            quarter          rbindlist              rleid 
                  TRUE               TRUE               TRUE               TRUE 
                rleidv             rollup              rowid             rowidv 
                  TRUE               TRUE               TRUE               TRUE 
                second                set        setalloccol            setattr 
                  TRUE               TRUE               TRUE               TRUE 
           setcolorder              setDF      setdroplevels              setDT 
                  TRUE               TRUE               TRUE               TRUE 
          setDTthreads           setindex          setindexv             setkey 
                  TRUE               TRUE               TRUE               TRUE 
               setkeyv          setnafill           setnames setNumericRounding 
                  TRUE               TRUE               TRUE               TRUE 
              setorder          setorderv              shift        shouldPrint 
                  TRUE               TRUE               TRUE               TRUE 
                    SJ        substitute2             tables    test.data.table 
                  TRUE               TRUE               TRUE               TRUE 
             timetaken          transpose         truelength          tstrsplit 
                  TRUE               TRUE               TRUE               TRUE 
               uniqueN     update_dev_pkg               wday               week 
                  TRUE               TRUE               TRUE               TRUE 
                  yday               year            yearmon            yearqtr 
                  TRUE               TRUE               TRUE               TRUE 

### Examples

There are 67 help pages with examples, from 67 (100.00 %).

### NEWS

The package has NEWS file and it is current.

### License

The package uses .

# Installation environment

## System Info

| Field          | Value                   |
|:---------------|:------------------------|
| Image          | rhub/ref-image          |
| OS             | Ubuntu 24.04.1 LTS      |
| Platform       | x86_64-pc-linux-gnu     |
| System         | x86_64, linux-gnu       |
| Execution Time | 2025-02-17 04:55:41 UTC |

**System information**. Table about the system used to check the
package.

## R Session Info

Information about the R environment and capabilities:

    R version 4.4.2 (2024-10-31)
    Platform: x86_64-pc-linux-gnu
    Running under: Ubuntu 24.04.1 LTS

    Matrix products: default
    BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
    LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0

    locale:
     [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       
     [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   
     [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          
    [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   

    time zone: UTC
    tzcode source: system (glibc)

    attached base packages:
    [1] tools     stats     graphics  grDevices utils     datasets  methods  
    [8] base     

    loaded via a namespace (and not attached):
     [1] riskreports_0.0.0.9003 compiler_4.4.2         fastmap_1.2.0         
     [4] cli_3.6.4              htmltools_0.5.8.1      yaml_2.3.10           
     [7] rmarkdown_2.29         knitr_1.49             jsonlite_1.8.9        
    [10] xfun_0.50              digest_0.6.37          rlang_1.1.5           
    [13] evaluate_1.0.3        

Platform used:

       OS.type   file.sep dynlib.ext        GUI     endian    pkgType   path.sep 
        "unix"        "/"      ".so"      "X11"   "little"   "source"        ":" 
        r_arch 
            "" 

R’s capabilities:

           jpeg         png        tiff       tcltk         X11        aqua 
           TRUE        TRUE        TRUE        TRUE       FALSE       FALSE 
       http/ftp     sockets      libxml        fifo      cledit       iconv 
           TRUE        TRUE       FALSE        TRUE       FALSE        TRUE 
            NLS       Rprof     profmem       cairo         ICU long.double 
           TRUE        TRUE        TRUE        TRUE        TRUE        TRUE 
        libcurl 
           TRUE 

External software:

                                                         zlib 
                                                        "1.3" 
                                                        bzlib 
                                         "1.0.8, 13-Jul-2019" 
                                                           xz 
                                                      "5.4.5" 
                                                   libdeflate 
                                                       "1.19" 
                                                         PCRE 
                                           "10.42 2022-12-11" 
                                                          ICU 
                                                       "74.2" 
                                                          TRE 
                                    "TRE 0.8.0 R_fixes (BSD)" 
                                                        iconv 
                                                 "glibc 2.39" 
                                                     readline 
                                                        "8.2" 
                                                         BLAS 
    "/usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3" 

Graphics external software:

                       cairo                  cairoFT                    pango 
                    "1.18.0"                       ""                 "1.52.1" 
                      libpng                     jpeg                  libtiff 
                    "1.6.43"                    "8.0" "LIBTIFF, Version 4.5.1" 

Numerical characteristics of the machine:

                   double.eps            double.neg.eps               double.xmin 
                 2.220446e-16              1.110223e-16             2.225074e-308 
                  double.xmax               double.base             double.digits 
                1.797693e+308              2.000000e+00              5.300000e+01 
              double.rounding              double.guard         double.ulp.digits 
                 5.000000e+00              0.000000e+00             -5.200000e+01 
        double.neg.ulp.digits           double.exponent            double.min.exp 
                -5.300000e+01              1.100000e+01             -1.022000e+03 
               double.max.exp               integer.max               sizeof.long 
                 1.024000e+03              2.147484e+09              8.000000e+00 
              sizeof.longlong         sizeof.longdouble            sizeof.pointer 
                 8.000000e+00              1.600000e+01              8.000000e+00 
                sizeof.time_t            longdouble.eps        longdouble.neg.eps 
                 8.000000e+00              1.084202e-19              5.421011e-20 
            longdouble.digits       longdouble.rounding          longdouble.guard 
                 6.400000e+01              5.000000e+00              0.000000e+00 
        longdouble.ulp.digits longdouble.neg.ulp.digits       longdouble.exponent 
                -6.300000e+01             -6.400000e+01              1.500000e+01 
           longdouble.min.exp        longdouble.max.exp 
                -1.638200e+04              1.638400e+04 

Random number generation process:

    [1] "Mersenne-Twister" "Inversion"        "Rejection"       

## Information about the environment

Environmental and options variables affect how package checks and
software in it might behave. Here is the environmental variables when
running this report

    _R_CHECK_SYSTEM_CLOCK_
                            FALSE
    ACCEPT_EULA             Y
    ACTIONS_RUNNER_ACTION_ARCHIVE_CACHE
                            /opt/actionarchivecache
    AGENT_TOOLSDIRECTORY    /opt/hostedtoolcache
    ANDROID_HOME            /usr/local/lib/android/sdk
    ANDROID_NDK             /usr/local/lib/android/sdk/ndk/27.2.12479018
    ANDROID_NDK_HOME        /usr/local/lib/android/sdk/ndk/27.2.12479018
    ANDROID_NDK_LATEST_HOME
                            /usr/local/lib/android/sdk/ndk/27.2.12479018
    ANDROID_NDK_ROOT        /usr/local/lib/android/sdk/ndk/27.2.12479018
    ANDROID_SDK_ROOT        /usr/local/lib/android/sdk
    ANT_HOME                /usr/share/ant
    AZURE_EXTENSION_DIR     /opt/az/azcliextensions
    BOOTSTRAP_HASKELL_NONINTERACTIVE
                            1
    BUNDLE_EXT              linux-amd64.deb
    CHROME_BIN              /usr/bin/google-chrome
    CHROMEWEBDRIVER         /usr/local/share/chromedriver-linux64
    CI                      true
    CONDA                   /usr/share/miniconda
    DEBIAN_FRONTEND         noninteractive
    DENO_DOM_PLUGIN         /opt/quarto/bin/tools/x86_64/deno_dom/libplugin.so
    DENO_NO_UPDATE_CHECK    1
    DENO_TLS_CA_STORE       system,mozilla
    DEPLOYMENT_BASEPATH     /opt/runner
    DOTNET_MULTILEVEL_LOOKUP
                            0
    DOTNET_NOLOGO           1
    DOTNET_SKIP_FIRST_TIME_EXPERIENCE
                            1
    EDGEWEBDRIVER           /usr/local/share/edge_driver
    EDITOR                  vi
    GECKOWEBDRIVER          /usr/local/share/gecko_driver
    GHCUP_INSTALL_BASE_PREFIX
                            /usr/local
    GITHUB_ACTION           __run
    GITHUB_ACTION_REF       
    GITHUB_ACTION_REPOSITORY
                            
    GITHUB_ACTIONS          true
    GITHUB_ACTOR            llrs-roche
    GITHUB_ACTOR_ID         185338939
    GITHUB_API_URL          https://api.github.com
    GITHUB_BASE_REF         
    GITHUB_ENV              /home/runner/work/_temp/_runner_file_commands/set_env_24251222-dc77-41e6-994a-03413e3e9f5f
    GITHUB_EVENT_NAME       schedule
    GITHUB_EVENT_PATH       /home/runner/work/_temp/_github_workflow/event.json
    GITHUB_GRAPHQL_URL      https://api.github.com/graphql
    GITHUB_HEAD_REF         
    GITHUB_JOB              main
    GITHUB_OUTPUT           /home/runner/work/_temp/_runner_file_commands/set_output_24251222-dc77-41e6-994a-03413e3e9f5f
    GITHUB_PATH             /home/runner/work/_temp/_runner_file_commands/add_path_24251222-dc77-41e6-994a-03413e3e9f5f
    GITHUB_REF              refs/heads/main
    GITHUB_REF_NAME         main
    GITHUB_REF_PROTECTED    false
    GITHUB_REF_TYPE         branch
    GITHUB_REPOSITORY       pharmaR/pharmapkgs
    GITHUB_REPOSITORY_ID    798326913
    GITHUB_REPOSITORY_OWNER
                            pharmaR
    GITHUB_REPOSITORY_OWNER_ID
                            42115094
    GITHUB_RETENTION_DAYS   90
    GITHUB_RUN_ATTEMPT      1
    GITHUB_RUN_ID           13362121460
    GITHUB_RUN_NUMBER       94
    GITHUB_SERVER_URL       https://github.com
    GITHUB_SHA              b25eba80408a4e3db53f5896d441ff2685638e62
    GITHUB_STATE            /home/runner/work/_temp/_runner_file_commands/save_state_24251222-dc77-41e6-994a-03413e3e9f5f
    GITHUB_STEP_SUMMARY     /home/runner/work/_temp/_runner_file_commands/step_summary_24251222-dc77-41e6-994a-03413e3e9f5f
    GITHUB_TRIGGERING_ACTOR
                            llrs-roche
    GITHUB_WORKFLOW         Update package repositories
    GITHUB_WORKFLOW_REF     pharmaR/pharmapkgs/.github/workflows/update-repos.yml@refs/heads/main
    GITHUB_WORKFLOW_SHA     b25eba80408a4e3db53f5896d441ff2685638e62
    GITHUB_WORKSPACE        /home/runner/work/pharmapkgs/pharmapkgs
    GOROOT_1_21_X64         /opt/hostedtoolcache/go/1.21.13/x64
    GOROOT_1_22_X64         /opt/hostedtoolcache/go/1.22.12/x64
    GOROOT_1_23_X64         /opt/hostedtoolcache/go/1.23.6/x64
    GRADLE_HOME             /usr/share/gradle-8.12.1
    HOME                    /home/runner
    HOMEBREW_CLEANUP_PERIODIC_FULL_DAYS
                            3650
    HOMEBREW_NO_AUTO_UPDATE
                            1
    ImageOS                 ubuntu24
    ImageVersion            20250209.1.0
    INVOCATION_ID           d5e471a50fd941d195319ae0e3c382bb
    JAVA_HOME               /usr/lib/jvm/temurin-17-jdk-amd64
    JAVA_HOME_11_X64        /usr/lib/jvm/temurin-11-jdk-amd64
    JAVA_HOME_17_X64        /usr/lib/jvm/temurin-17-jdk-amd64
    JAVA_HOME_21_X64        /usr/lib/jvm/temurin-21-jdk-amd64
    JAVA_HOME_8_X64         /usr/lib/jvm/temurin-8-jdk-amd64
    JOURNAL_STREAM          8:6772
    LANG                    C.UTF-8
    LD_LIBRARY_PATH         /opt/R/4.4.2/lib/R/lib:/usr/local/lib:/usr/lib/x86_64-linux-gnu:/usr/lib/jvm/temurin-17-jdk-amd64/lib/server:/opt/R/4.4.2/lib/R/lib:/usr/local/lib:/usr/lib/x86_64-linux-gnu:/usr/lib/jvm/temurin-17-jdk-amd64/lib/server
    LN_S                    ln -s
    LOGGER_LOG_LEVEL        DEBUG
    MAKE                    make
    MEMORY_PRESSURE_WATCH   /sys/fs/cgroup/system.slice/runner-provisioner.service/memory.pressure
    MEMORY_PRESSURE_WRITE   c29tZSAyMDAwMDAgMjAwMDAwMAA=
    NOT_CRAN                true
    NVM_DIR                 /home/runner/.nvm
    PAGER                   /usr/bin/pager
    PATH                    /snap/bin:/home/runner/.local/bin:/opt/pipx_bin:/home/runner/.cargo/bin:/home/runner/.config/composer/vendor/bin:/usr/local/.ghcup/bin:/home/runner/.dotnet/tools:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
    PERFLOG_LOCATION_SETTING
                            RUNNER_PERFLOG
    PHARMAPKGS_EXCLUDED_METRICS
                            assess_covr_coverage,assess_r_cmd_check
    PHARMAPKGS_LIMIT        1000
    PHARMAPKGS_PLATFORM     ubuntu-22.04
    PHARMAPKGS_R_VERSION    4.4
    PHARMAPKGS_REMOTE_REPO
                            https://cloud.r-project.org/
    PIPX_BIN_DIR            /opt/pipx_bin
    PIPX_HOME               /opt/pipx
    PKGCACHE_HTTP_VERSION   2
    PKGLOAD_PARENT_TEMPDIR
                            /tmp/RtmpjxUEUq
    POWERSHELL_DISTRIBUTION_CHANNEL
                            GitHub-Actions-ubuntu24
    PROCESSX_PS1bd6333a6894_1739768135
                            YES
    PWD                     /home/runner/work/pharmapkgs/pharmapkgs
    QUARTO_BIN_PATH         /opt/quarto/bin
    QUARTO_DENO             /opt/quarto/bin/tools/x86_64/deno
    QUARTO_DOCUMENT_PATH    /home/runner/work/pharmapkgs/pharmapkgs/inst/report
    QUARTO_PROFILE          
    QUARTO_PROJECT_DIR      /home/runner/work/pharmapkgs/pharmapkgs
    QUARTO_PROJECT_ROOT     /home/runner/work/pharmapkgs/pharmapkgs
    QUARTO_ROOT             /
    QUARTO_SHARE_PATH       /opt/quarto/share
    R_ARCH                  
    R_BROWSER               xdg-open
    R_BZIPCMD               /usr/bin/bzip2
    R_DOC_DIR               /opt/R/4.4.2/lib/R/doc
    R_GZIPCMD               /usr/bin/gzip
    R_HOME                  /opt/R/4.4.2/lib/R
    R_INCLUDE_DIR           /opt/R/4.4.2/lib/R/include
    R_LIB_FOR_PAK           /opt/R/4.4.2/lib/R/site-library
    R_LIBS_SITE             /opt/R/4.4.2/lib/R/site-library
    R_LIBS_USER             /home/runner/work/_temp/Library
    R_PAPERSIZE             letter
    R_PAPERSIZE_USER        letter
    R_PDFVIEWER             /usr/bin/xdg-open
    R_PLATFORM              x86_64-pc-linux-gnu
    R_PRINTCMD              /usr/bin/lpr
    R_RD4PDF                times,inconsolata,hyper
    R_SESSION_TMPDIR        /tmp/RtmpngjTMs
    R_SHARE_DIR             /opt/R/4.4.2/lib/R/share
    R_STRIP_SHARED_LIB      strip --strip-unneeded
    R_STRIP_STATIC_LIB      strip --strip-debug
    R_TEXI2DVICMD           /usr/bin/texi2dvi
    R_UNZIPCMD              /usr/bin/unzip
    R_ZIPCMD                /usr/bin/zip
    RUNNER_ARCH             X64
    RUNNER_ENVIRONMENT      github-hosted
    RUNNER_NAME             GitHub Actions 16
    RUNNER_OS               Linux
    RUNNER_PERFLOG          /home/runner/perflog
    RUNNER_TEMP             /home/runner/work/_temp
    RUNNER_TOOL_CACHE       /opt/hostedtoolcache
    RUNNER_TRACKING_ID      github_84602b48-6461-4335-a438-3ca558c890e8
    RUNNER_USER             runner
    RUNNER_WORKSPACE        /home/runner/work/pharmapkgs
    SED                     /usr/bin/sed
    SELENIUM_JAR_PATH       /usr/share/java/selenium-server.jar
    SGX_AESM_ADDR           1
    SHLVL                   1
    STATS_D                 true
    STATS_D_D               true
    STATS_D_TC              true
    STATS_EXT               true
    STATS_EXTP              https://provjobdprod.z13.web.core.windows.net/settings/provjobdsettings-latest/provjobd.data
    STATS_PIP               false
    STATS_RDCL              true
    STATS_TRP               true
    STATS_UE                true
    STATS_V3PS              true
    STATS_VMD               true
    STATS_VMFE              true
    SWIFT_PATH              /usr/share/swift/usr/bin
    SYSTEMD_EXEC_PID        830
    TAR                     /usr/bin/tar
    TZ                      UTC
    USER                    runner
    VCPKG_INSTALLATION_ROOT
                            /usr/local/share/vcpkg
    XDG_CONFIG_HOME         /home/runner/.config
    XDG_RUNTIME_DIR         /run/user/1001

These are the options set to generate the report:

    $add.smooth
    [1] TRUE

    $bitmapType
    [1] "cairo"

    $browser
    [1] "xdg-open"

    $browserNLdisabled
    [1] FALSE

    $callr.condition_handler_cli_message
    function (msg) 
    {
        custom_handler <- getOption("cli.default_handler")
        if (is.function(custom_handler)) {
            custom_handler(msg)
        }
        else {
            cli_server_default(msg)
        }
    }
    <bytecode: 0x55732fe64910>
    <environment: namespace:cli>

    $catch.script.errors
    [1] FALSE

    $CBoundsCheck
    [1] FALSE

    $check.bounds
    [1] FALSE

    $citation.bibtex.max
    [1] 1

    $continue
    [1] "+ "

    $contrasts
            unordered           ordered 
    "contr.treatment"      "contr.poly" 

    $covr.record_tests
    [1] TRUE

    $defaultPackages
    [1] "datasets"  "utils"     "grDevices" "graphics"  "stats"     "methods"  

    $demo.ask
    [1] "default"

    $deparse.cutoff
    [1] 60

    $device
    function (width = 7, height = 7, ...) 
    {
        grDevices::pdf(NULL, width, height, ...)
    }
    <bytecode: 0x55732f54cb50>
    <environment: namespace:knitr>

    $device.ask.default
    [1] FALSE

    $digits
    [1] 7

    $dvipscmd
    [1] "dvips"

    $echo
    [1] FALSE

    $editor
    [1] "vi"

    $encoding
    [1] "native.enc"

    $example.ask
    [1] "default"

    $expressions
    [1] 5000

    $help.search.types
    [1] "vignette" "demo"     "help"    

    $help.try.all.packages
    [1] FALSE

    $htmltools.preserve.raw
    [1] TRUE

    $HTTPUserAgent
    [1] "R/4.4.2 R (4.4.2 x86_64-pc-linux-gnu x86_64 linux-gnu) on GitHub Actions"

    $internet.info
    [1] 2

    $keep.parse.data
    [1] TRUE

    $keep.parse.data.pkgs
    [1] FALSE

    $keep.source
    [1] FALSE

    $keep.source.pkgs
    [1] FALSE

    $knitr.in.progress
    [1] TRUE

    $locatorBell
    [1] TRUE

    $mailer
    [1] "mailto"

    $matprod
    [1] "default"

    $max.contour.segments
    [1] 25000

    $max.print
    [1] 99999

    $menu.graphics
    [1] TRUE

    $na.action
    [1] "na.omit"

    $Ncpus
    [1] 1

    $nwarnings
    [1] 50

    $OutDec
    [1] "."

    $pager
    [1] "/opt/R/4.4.2/lib/R/bin/pager"

    $papersize
    [1] "letter"

    $PCRE_limit_recursion
    [1] NA

    $PCRE_study
    [1] FALSE

    $PCRE_use_JIT
    [1] TRUE

    $pdfviewer
    [1] "/usr/bin/xdg-open"

    $pkgType
    [1] "source"

    $printcmd
    [1] "/usr/bin/lpr"

    $prompt
    [1] "> "

    $repos
                          CRAN 
    "https://cran.rstudio.com" 

    $rl_word_breaks
    [1] " \t\n\"\\'`><=%;,|&{()}"

    $rlang_trace_top_env
    <environment: R_GlobalEnv>

    $scipen
    [1] 0

    $show.coef.Pvalues
    [1] TRUE

    $show.error.messages
    [1] TRUE

    $show.signif.stars
    [1] TRUE

    $showErrorCalls
    [1] TRUE

    $showNCalls
    [1] 50

    $showWarnCalls
    [1] FALSE

    $str
    $str$strict.width
    [1] "no"

    $str$digits.d
    [1] 3

    $str$vec.len
    [1] 4

    $str$list.len
    [1] 99

    $str$deparse.lines
    NULL

    $str$drop.deparse.attr
    [1] TRUE

    $str$formatNum
    function (x, ...) 
    format(x, trim = TRUE, drop0trailing = TRUE, ...)
    <environment: 0x55732f33c520>


    $str.dendrogram.last
    [1] "`"

    $texi2dvi
    [1] "/usr/bin/texi2dvi"

    $tikzMetricsDictionary
    [1] "validation_report_data.table_v1.16.4-tikzDictionary"

    $timeout
    [1] 60

    $try.outFile
    A connection with                    
    description ""      
    class       "file"  
    mode        "w+b"   
    text        "binary"
    opened      "opened"
    can read    "yes"   
    can write   "yes"   

    $ts.eps
    [1] 1e-05

    $ts.S.compat
    [1] FALSE

    $unzip
    [1] "/usr/bin/unzip"

    $useFancyQuotes
    [1] FALSE

    $verbose
    [1] FALSE

    $warn
    [1] 0

    $warning.length
    [1] 1000

    $warnPartialMatchArgs
    [1] FALSE

    $warnPartialMatchAttr
    [1] FALSE

    $warnPartialMatchDollar
    [1] FALSE

    $width
    [1] 80
