<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.13">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>validation_report_data.table_v1.16.4 â€“ pharmapkgs: riskreports</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-html/quarto.js" type="module"></script>
<script src="../../site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-d191db7fc3c3d9a11a0015f971215285.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-dark-d191db7fc3c3d9a11a0015f971215285.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap-4155540381b49a91af01653b41d53e74.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../../site_libs/bootstrap/bootstrap-dark-550fc17191dd18cbea4970ed4e279535.min.css" rel="prefetch" append-hash="true" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


</head>

<body class="nav-fixed fullcontent">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">pharmapkgs: riskreports</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/pharmaR/pharmapkgs"> <i class="bi bi-github" role="img" aria-label="GitHub repo">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block"></header>




<section id="validation-report---data.table1.16.4" class="level1">
<h1>Validation Report - data.table@1.16.4</h1>
</section>
<section id="context" class="level1">
<h1>Context</h1>
<p>This report is fully automated and builds on rhub/ref-image image. Documents the installation of this package on an open source R environment, focusing on:</p>
<ul>
<li>Installation environment description</li>
<li>Testing coverage</li>
</ul>
<p>It is limited to assess whether unit tests and documentation are present and can execute without error. An assessment would be required that the tests and documentation are meaningful.</p>
</section>
<section id="package-data.table" class="level1">
<h1>Package data.table</h1>
<section id="metric-based-risk-assessment" class="level2">
<h2 class="anchored" data-anchor-id="metric-based-risk-assessment">Metric based risk assessment</h2>
<p>The following metrics are derived from the <code>riskmetric</code> R package.</p>
<table class="caption-top table">
<tbody>
<tr class="odd">
<td style="text-align: left;">has_news</td>
<td style="text-align: left;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">news_current</td>
<td style="text-align: left;">TRUE</td>
</tr>
<tr class="odd">
<td style="text-align: left;">exported_namespace</td>
<td style="text-align: left;">120</td>
</tr>
<tr class="even">
<td style="text-align: left;">has_vignettes</td>
<td style="text-align: left;">25</td>
</tr>
<tr class="odd">
<td style="text-align: left;">export_help</td>
<td style="text-align: left;">120</td>
</tr>
<tr class="even">
<td style="text-align: left;">has_website</td>
<td style="text-align: left;">TRUE</td>
</tr>
<tr class="odd">
<td style="text-align: left;">has_maintainer</td>
<td style="text-align: left;">Tyson Barrett <a href="mailto:t.barrett88@gmail.com" class="email">t.barrett88@gmail.com</a></td>
</tr>
<tr class="even">
<td style="text-align: left;">bugs_status</td>
<td style="text-align: left;">subscript out of bounds</td>
</tr>
<tr class="odd">
<td style="text-align: left;">has_source_control</td>
<td style="text-align: left;">https://github.com/Rdatatable/data.table</td>
</tr>
<tr class="even">
<td style="text-align: left;">has_bug_reports_url</td>
<td style="text-align: left;">1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">downloads_1yr</td>
<td style="text-align: left;">9897084</td>
</tr>
<tr class="even">
<td style="text-align: left;">reverse_dependencies</td>
<td style="text-align: left;">1958</td>
</tr>
<tr class="odd">
<td style="text-align: left;">has_examples</td>
<td style="text-align: left;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">dependencies</td>
<td style="text-align: left;">1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">license</td>
<td style="text-align: left;">MPL-2.0 | file LICENSE</td>
</tr>
</tbody>
</table>
<p><strong>Package general assessment:</strong> Coverage, check results, size, download the last year, reverse dependencies and number of dependencies.</p>
</section>
<section id="dependencies" class="level2">
<h2 class="anchored" data-anchor-id="dependencies">Dependencies</h2>
<p>Overall the package has these dependencies:</p>
<table class="caption-top table">
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: left;">package</th>
<th style="text-align: left;">type</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">1</td>
<td style="text-align: left;">methods</td>
<td style="text-align: left;">Imports</td>
</tr>
</tbody>
</table>
<p><strong>Package dependencies</strong></p>
</section>
<section id="reverse-dependencies" class="level2">
<h2 class="anchored" data-anchor-id="reverse-dependencies">Reverse dependencies</h2>
<p>Overall the package has 1958 reverse dependencies:</p>
<p>Ac3net, accessibility, acdcR, Achilles, acoRn, actel, ActivePathways, ActivityIndex, actuaRE, ADAMgui, adductomicsR, AdhereR, AdhereRViz, ADImpute, adjustedCurves, AeRobiology, AF, AFM, africamonitor, agcounts, AggregateR, agriutilities, ahw, AICcPermanova, airship, akc, akiFlagger, aLFQ, alpaca, AlphaBeta, amadeus, amapGeocode, AMAPVox, amerifluxr, AMIM, amplican, AMR, amt, Anaconda, analyzer, animint2, antaresEditObject, antaresProcessing, antaresRead, antaresViz, AnxietySleep, aopdata, aorsf, AOV1R, APCI, APIS, aplotExtra, AQEval, aqp, arcgisgeocode, arcgisutils, arctools, arf, artMS, asciiSetupReader, ascotraceR, aslib, ASpli, ASRgenomics, assertable, atime, AtmChile, atSNP, attrib, AUCell, aum, autoCovariateSelection, autonewsmd, autonomics, AutoPlots, autostats, autostsm, aweSOM, bagged.outliertrees, bakR, bambu, BANDITS, Banksy, baseballr, basecallQC, BatchExperiments, BatchJobs, BatchQC, batchtools, BayesianMCPMod, bayesLife, BayesLN, bayesMeanScale, bayesMig, bayesPop, BayesSUR, bayesTFR, bbk, BBmisc, bbmix, bbotk, bcputility, bdots, bea.R, BEARscc, BEclear, BEDMatrix, bedr, BeeGUTS, behavr, benford.analysis, bfw, BGData, BgeeCall, BgeeDB, BGmisc, BiasCorrector, biblionetwork, bife, BIGL, bigreadr, bigsnpr, bigstatsr, bikeshare14, binancer, binomialRF, binsegRcpp, biobroom, BiocParallel, biomartr, BIOMASS, BioNAR, biopixR, bioregion, birankr, biscuiteer, bkmrhat, BKTR, BLA, blorr, BMisc, bnbc, BootstrapQTL, BPRMeth, bRacatus, brainGraph, branchpointer, brandwatchR, bratteli, bread, BREADR, bridger2, BRINDA, brmsmargins, brpop, bruceR, bsitar, bsseq, BTM, BTSPAS, BTYDplus, bupaR, BuyseTest, bysykkel, c3, CAGEr, CALF, calmr, CamelUp, camtrapR, CaMutQC, CareDensity, caretEnsemble, cartogramR, casebase, CAST, CATALYST, catcont, categoryEncodings, CatEncoders, CatsCradle, causalCmprsk, CausalGPS, causalOT, cbl, CBRT, ccdR, ccmap, ccostr, cdid, cdparcoord, ceas, celda, CellBarcode, cellbaseR, CelliD, cellKey, CellNOptR, CEMiTool, Census2016, cercospoRa, Certara.NLME8, Certara.R, Certara.RsNLME, Certara.RsNLME.ModelBuilder, Certara.VPCResults, Certara.Xpose.NLME, cffdrs, cfr, cglm, changepoint.influence, ChannelAttributionApp, cheapr, checkmate, chem.databases, ChemoSpecUtils, Chicago, chicane, chimeraviz, ChineseNames, ChIPexoQual, ChIPpeakAnno, chronicle, cicero, circhelp, citationchaser, citecorp, ciw, clam, classifierplots, cleanTS, ClickHouseHTTP, clickstream, climate, clinDataReview, clinspacy, clinUtils, CluMP, ClustAssess, Clustering, clustifyr, clustra, CLVTools, cmapR, cmsaf, CNVRanger, CNVScope, COCOA, coder, CoDiNA, coffee, cohorts, coinmarketcapr, collapse, coloc, colocPropTest, colorDF, colorednoise, Colossus, comorbidity, CompareCausalNetworks, compareDF, COMPASS, ComplexUpset, ComPrAn, compSPOT, CONFESS, CoNI, conleyreg, consensusDE, consortr, ConsReg, constellation, constructive, container, CoOL, cooltools, CooRTweet, CopernicusDEM, cophescan, coRdon, CoreGx, CornerstoneR, corporaexplorer, corpustools, corrcoverage, correspondenceTables, CoSMoS, COTAN, counterfactuals, countries, coveffectsplot, COVID19, covid19br, covid19india, covidnor, cppRouting, CRE, creditmodel, crew, crfsuite, CRISPRseek, CRMetrics, CropScapeR, crosshap, crossmeta, crumble, cryptoQuotes, csa, csalert, csdata, csmaps, CSMES, cstidy, cstime, CSTools, csutil, csv, csvy, ctbi, ctmm, cTRAP, ctsem, ctxR, cubar, curtailment, cusum, cusumcharter, cvAUC, cvms, CWT, cxhull, cyclestreets, cylcop, cytoKernel, CytoML, D3partitionR, daiquiri, daiR, damr, dang, dartR.base, dartR.popgen, dartR.sim, dartR.spatial, dat, data.table.threads, data360r, DataCombine, dataCompare, dataCompareR, datadictionary, DataExplorer, dataframeexplorer, datamods, dataMojo, datapasta, dataprep, dataPreparation, dataRetrieval, datarobot, datasetjson, DataSpaceR, DATAstudio, dataverifyr, datawizard, datazoom.amazonia, dateutils, dbi.table, dblr, dbnR, dbWebForms, ddtlcm, DeclareDesign, decompTumor2Sig, deconvR, dedupewider, deepdive, deeptrafo, DEFormats, DegNorm, delayed, deltaccd, DemoKin, denvax, DeProViR, DEScan2, DescTools, designit, DESpace, detectRUNS, DEWSeq, dfmeta, DGM, DIAlignR, diathor, diceplot, did2s, Diderot, DiDforBigData, didimputation, DIDmultiplegtDYN, difconet, DifferentialRegulation, DiscoRhythm, DiSCos, discSurv, DisImpact, disk.frame, distillML, DIZtools, DIZutils, DMCFB, dMrs, do, DominoEffect, doremi, DoubleML, DQAgui, DQAstats, dQTG.seq, drake, DramaAnalysis, dreamlet, drgee, dropR, droptest, dsos, DTAT, dtplyr, DTSg, dtts, DTwrappers, DTwrappers2, dumbbell, Durga, dvmisc, dynamite, DYNATE, dynsurv, DysPIA, E4tools, Eagle, easybio, easycsv, easyr, easysurv, eatGADS, eatTools, EBPRS, eclust, ecocomDP, ecoregime, ecospace, ecotox, edeaR, editbl, edl, EdSurvey, eeptools, efdm, effClust, EGM, EHR, EIAapi, EIX, eixport, electionsBR, ELMER, eltr, EmiR, EmissV, enderecobr, endogeneity, endoSwitch, enmSdmX, ensr, envalysis, enviGCMS, eoR, epanetReader, Epi, epialleleR, epiCleanr, EpiCompare, epidm, EpiForsk, epigraHMM, EpiMix, EpiNow2, epiomics, epistasisGA, epiworldR, eplusr, epm, epwshiftr, equateMultiple, eSIR, estadistica, EthSEQ, etm, etwfe, eurocordexr, eurodata, europeanaR, eurostat, evapoRe, eventglm, eventstudyr, evoper, EWCE, exiftoolr, explainer, exploratory, expowo, expss, exvatools, eyeRead, eyeTrackR, ezr, fabricatr, facmodCS, facmodTS, FACT, factor256, factR, faers, faersquarterlydata, FairMclus, familiar, FAOSTAT, farrell, fastai, fastdid, fastDummies, fasterRaster, fastLink, fastreeR, FastRet, fastRhockey, fastText, fastverse, fastVoteR, fbRads, fcfdr, FCO, fdq, FeatureImpCluster, fedmatch, fedstatAPIr, fedup, felp, fgdr, Fgmutils, fgsea, fhircrackr, FIESTA, FIESTAutils, filecacher, finbif, finch, finnsurveytext, fishmethods, fishpond, fitbitViz, FitHiC, fixest, flextable, flightsbr, flippant, FLOPART, florabr, flowGraph, flowml, flowTraceR, flowWorkspace, flps, Fluidigm, FMAT, fmeffects, FOCI, forceplate, forecastML, forestError, forImage, formatdown, formulaic, FORTLS, fossilbrush, FourWayHMM, fplot, fplyr, FRASER, fruclimadapt, fsbrain, fst, FuncDiv, funchir, future.tests, FuzzyDBScan, fwlplot, fy, GALLO, GARCOM, gasfluxes, gatom, GaussSuppression, gbp, gcite, gde, gDRcore, gDRimport, gDRutils, gdxdt, gemma.R, gems, GeneGeneInteR, geneplast, GENESIS, GeneStructureTools, geneXtendeR, genomation, GenomAutomorphism, GenomeInfoDb, genomic.autocorr, GenomicDataCommons, GenomicDistributions, GenomicInteractions, GenomicScores, GenomicTools.fileHandler, GenomicTuples, GenoScan, GenVisR, geobr, geocausal, geocodebr, GeoMongo, GeomxTools, geoperu, GEOquery, GeoTcgaData, GeoThinneR, germinationmetrics, getDTeval, GFE, gfer, gfoRmula, gfoRmulaICE, ggbrain, ggcyto, ggdmc, ggetho, ggfacto, ggfigdone, ggfixest, GGIR, GGIRread, ggkegg, ggmugs, ggperiodic, ggPMX, ggpval, ggquickeda, ggsom, ggswissmaps, ggTimeSeries, GHap, githubinstall, gittargets, GladiaTOX, GmicR, GMMAT, gmoTree, gnrprod, goat, gompertztrunc, googleCloudVisionR, googleVis, GOTHiC, gpboost, grandR, GRaNIE, grattan, grattanInflators, greatR, GREENeR, greport, GRIN2, growthcleanr, GRShiny, GSALightning, gsDesign, gsDesign2, GSODR, GSVA, gtfs2emis, gtfs2gps, gtfsio, gtfsrouter, gtfstools, GTFSwizard, GUIDEseq, gunsales, gustave, gwaRs, gwasforest, GWASinspector, GWASTools, gwid, h2o, HACSim, HaDeX, hagis, hahmmr, hal9001, haldensify, handlr, harmony, hbamr, hBayesDM, HCD, HCR, hdd, hdpGLM, healthdb, healthyAddress, heatwaveR, heims, hereR, hermiter, HERON, hesim, heuristicsmineR, HicAggR, HiCBricks, HiCcompare, HiCDCPlus, HiCDOC, hicp, highfrequency, hilldiv, himach, hisse, Hmisc, HMMcopy, hoardeR, HodgesTools, hollr, hoopR, HospitalNetwork, hot.deck, HPLB, hR, hsrecombi, htetree, httk, hudr, hurricaneexposure, hutils, hutilscpp, HVT, hwig, hydroloom, hydropeak, hyfo, hypervolume, HYPEtools, i2extras, IBRtools, iCAMP, ICAMS, iCellR, IceSat2R, iCNV, ideanet, IDEATools, IDmining, ie2misc, ie2miscdata, ieegio, iemisctext, ifCNVR, iGC, iimi, imabc, imcRtools, imf.data, iml, immunarch, immuneSIM, implicitMeasures, importinegi, imputeMulti, inbreedR, IncDTW, incidence2, ineAtlas, inferr, InflectSSP, influential, Information, INLAspacetime, inplace, installr, insurancerating, InteractiveComplexHeatmap, InterCellar, interface, intervalaverage, inTrees, IOHanalyzer, ipeaplot, ips, irace, iraceplot, irg, IrregLong, ISAnalytics, ISCA, iscoCrosswalks, isdparser, IsoBayes, isobxr, isotracer, iSTATS, itol.toolkit, itsadug, itscalledsoccer, ivdoctr, ivmte, ivtools, jab.adverse.reactions, jage, JointFPM, joinXL, journalabbr, joyn, jrSiCKLSNMF, JSmediation, jsmodule, jstable, jsTree, jubilee, JWileymisc, kalmanfilter, kangar00, kdry, kerastuneR, kesernetwork, kgen, kgraph, kibior, kim, kimfilter, KinSwingR, kisopenapi, KMD, kmeRtone, knfi, KnockoffScreen, koRpus, kosis, KPC, lab2clean, LabourMarketAreas, labourR, LACE, lacunr, Lahman, LambertW, lar, Largevars, latrend, lava, lavaSearch2, lay, lazytrade, LCMSQA, ldamatch, ldaPrototype, ldsr, leafR, leakyIV, LearningStats, leiden, lexicon, LexisNexisTools, lfe, lgr, lgrExtra, libbib, libr, lidR, lightgbm, LilRhino, limexhub, limorhyde, limorhyde2, lineagespot, LinkHD, lipidr, lisaClust, live, LKT, llama, LMMstar, LMPdata, lmtp, loadeR, LocalControl, loci2path, logiBin, LogicForest, LOLA, LOMAR, lookupTable, LOPART, lori, LPDynR, LSPFP, Luminescence, LWFBrook90R, lwqs, LymphoSeq, m5, Maaslin2, Macarron, maditr, maftools, MAGAR, magclass, MAGEE, maicChecks, MAICtools, MANOVA.RM, MantaID, manydist, mapi, marginaleffects, margins, mariner, marked, maser, MAST, matman, MatrixExtra, MatrixHMM, mau, maxaltall, maxent.ot, mbr, mbRes, mcauchyd, mcboost, medrxivr, Melissa, memisc, memochange, mergen, MesKit, metabinR, metabodecon, metabolomicsWorkbenchR, MetaComp, metaforest, metaGE, metagene2, MetAlyzer, metamer, metasnf, meteo, methimpute, methodical, MethReg, methrix, methylKit, MethylMix, methylPipe, methylscaper, metR, mggd, mgi.report.reader, MIC, miceFast, micemd, miceRanger, MicrobiotaProcess, microeco, MicroMoB, microseq, midfieldr, miesmuschel, miLineage, milr, MIMER, mimi, mind, minfi, MinimumDistance, minSNPs, mintyr, mipplot, MIRA, mirTarRnaSeq, miscset, missCompare, misty, mixgb, mixIndependR, mixvlmc, mknapsack, mlbplotR, mldr.resampling, mlexperiments, mllrnrs, mlr, mlr3, mlr3batchmark, mlr3benchmark, mlr3cluster, mlr3db, mlr3fairness, mlr3fda, mlr3filters, mlr3fselect, mlr3hyperband, mlr3inferr, mlr3learners, mlr3mbo, mlr3misc, mlr3oml, mlr3pipelines, mlr3resampling, mlr3shiny, mlr3spatial, mlr3spatiotempcv, mlr3summary, mlr3superlearner, mlr3torch, mlr3tuning, mlr3tuningspaces, mlr3verse, mlr3viz, mlrintermbo, mlrMBO, mlsurvlrnrs, mltools, mmaqshiny, mnem, mobileRNA, MOCHA, modACDC, ModCon, ModelMetrics, modelsummary, MOFA2, MoleculeExperiment, MouseFM, MPAC, mpactr, MplusAutomation, mpwR, mrgsolve, mrMLM, mrMLM.GUI, mrregression, MsBackendSql, mschart, msImpute, mslp, MSmix, MSnID, MSstats, MSstatsConvert, MSstatsLiP, MSstatsPTM, MSstatsShiny, MSstatsTMT, mstate, mstclustering, mstrio, mstudentd, mtb, mulea, multicrispr, multifamm, MultiFit, multiHiCcompare, multilevelcoda, multilevelTools, multinets, multistateQTL, multiWGCNA, multvardiv, MungeSumstats, muscat, musica, musicatk, mutationtypes, mutualinf, mvQuad, mvSUSY, mxnorm, myClim, naaccr, NACHO, nanotime, naptanr, nc, ndjson, neatR, neatRanges, neatStats, neo4r, NEONiso, neonOS, neonPlantEcology, neonUtilities, nestedcv, net4pg, netCoin, netgsa, NetLogoR, netSmooth, networkR, netZooR, neuroimaGene, nfl4th, nflfastR, nflplotR, nflreadr, nflseedR, NFLSimulatoR, nhppp, nlmixr2est, nlmixr2extra, nlpembeds, nlpred, NMcalc, NMdata, NMsim, nngeo, NNS, noctua, node2vec, noncompliance, nonmem2rx, normfluodbf, nosoi, npcurePK, nRegression, nullranges, numbat, o2geosocial, OBIC, obliqueRSF, occupationMeasurement, octad, octopus, odbr, oddsapiR, ofpetrial, ogrdbstats, OGRE, okxAPI, OlinkAnalyze, OmicNavigator, OMICsPCA, omicwas, ompr, OnboardClient, OncoSimulR, One4All, onlineforecast, onsr, openCyto, opendataformat, openEBGM, OpenML, OpenSpecy, opentripplanner, optiSel, orderanalyzer, ORFhunteR, ORFik, orgR, origami, origin, orthogene, osmclass, outliertree, OUTRIDER, overviewR, owdbr, owidR, paar, packageRank, padr, pagoda2, pairkat, PakPMICS2014Ch, PakPMICS2014HH, PakPMICS2014HL, PakPMICS2014Wm, pam, PAMpal, PAMscapes, pander, panelaggregation, PanelMatch, panelWranglR, panstarrs, PanViz, paradox, parallelMap, paramix, ParBayesianOptimization, parseRPDR, parSim, partCNV, partools, parttree, pathviewr, pathwayTMB, patternator, pbox, pcadapt, PCBS, PCMBase, PCMBaseCpp, PCRA, pcvr, pda, PDATK, pdynmc, PeakSegDisk, PeakSegOptimal, pedalfast.data, pedbp, Pedixplorer, pedquant, penaltyLearning, PepMapViz, pepr, pepStat, PGRdup, pgTools, phantasus, phantasusLite, PharmacoGx, pheble, phenofit, phenomis, PhenotypeSimulator, phers, phm, phonics, PhyloProfile, phyloseq, PhysicalActivity, pingers, pins, pinyin, pipeflow, pivottabler, pk4adi, pkggraph, pkgnet, PKPDsim, pksensi, PlanetNICFI, plinkQC, plmmr, plnr, plotgardener, plotHMM, plotly, pmartR, pmcalibration, pmml, pmparser, pmxpartab, pointblank, pointdensityP, poisFErobust, politeness, polle, polmineR, polypharmacy, pomp, POMS, poolfstat, pooling, popEpi, PopulateR, PortfolioAnalytics, portvine, posologyr, POSSA, potools, POUMM, power.transform, powerbiR, PP, ppmlasso, pram, pRecipe, precrec, predhy.GUI, prediction, PReMiuM, PreProcessRecordLinkage, preputils, PressPurt, preventr, previsionio, pricelevels, primer, prioriactions, prioritizr, prnsamplr, proActiv, processmapR, processpredictR, prodlim, ProjectTemplate, prolific.api, PRONE, PROSPER, Prostar, protGear, protti, prrd, prt, psborrow, PSF, psichomics, psidR, psmineR, psp, psycCleaning, PsychWordVec, ptable, ptairMS, pTITAN2, PTXQC, Publish, PureCN, puremoe, pureseqtmr, pvda, Pviz, pvLRT, qad, QCAcluster, qckitfastq, qdapTools, qgg, qreport, qs, qs2, qte, QTL.gCIMapping, QTL.gCIMapping.GUI, qtl2, QuartPAC, quickcheck, quickPlot, R.AlphA.Home, r2dii.match, R2DT, r2glmm, R3CPET, r3Cseq, r5r, R62S3, R6causal, RABR, RadEro, RadioGx, radlibs, RAINBOWR, RALSA, rAmCharts, RaMS, randnet, randomForestExplainer, randomForestSRC, randomNames, rangeMapper, RankAggSIgFUR, RapidoPGS, raqs, RARfreq, rassta, rasterDT, RAthena, ratios, Ravages, RawHummus, rbacon, RBaseX, rBayesianOptimization, rbi, rbi.helpers, rBiasCorrection, rbin, rBiopaxParser, Rblpapi, rblt, Rborist, rbtt, Rcan, RCAS, RcextTools, rchemo, RcisTarget, RcmdrPlugin.RiskDemo, RcmdrPlugin.WorldFlora, RCNA, rCNV, RcppGreedySetCover, rcprd, Rcrawler, rdbnomics, rdfp, rdhs, Rdiagnosislist, Rdice, RDML, rdracor, rdwd, ReactomeContentService4R, ReactomeGraph4R, read.gt3x, readNSx, readtext, realTimeloads, reappraised, rechaRge, reclin2, recometrics, recorder, RecordLinkage, recount3, redatam, Rediscover, RefBasedMI, regport, regtools, rehh, ReinforcementLearning, reinsureR, remiod, remotePARTS, rENA, REndo, replacer, repmis, reproducible, Require, rerddap, resemble, reservr, reslr, RESOLVE, respR, restatapi, ReSurv, rfaRm, rfars, rfisheries, RFplus, rfPred, rgbif, RGMQL, rhdf5client, RIbench, RiboCrypt, RiboDiPA, RiboProfiling, ricu, Rilostat, RImmPort, rintcal, rio, rIsing, riskRegression, RITCH, ritis, rlas, rlist, rllama, RLumShiny, RMassBank, rmcfs, rMIDAS, RNewsflow, rnpn, Rnvd3, roads, robCompositions, RobinCar, robis, robotoolbox, ROCket, rODE, rollinglda, rorcid, rotor, rPackedBar, rpivotTable, rprev, rqdatatable, rqPen, rreg, Rrepest, RSauceLabs, rSPARCS, rsparse, rstanarm, rsurveycto, rsyntax, RTCGA, RTCGAToolbox, rtmpt, RTN, RTNsurvival, rtop, RTransferEntropy, rtry, rtsdata, RTTWebClient, ruimtehol, runcharter, rush, rusquant, RVA, RWDataPlyr, rWikiPathways, RWildbook, rworkflows, RxnSim, rxode2, ryandexdirect, rzentra, RZooRoH, s3fs, S4Vectors, saeHB.twofold, SafeVote, salesforcer, samadb, sampcompR, samplingin, sangeranalyseR, sanityTracker, sansa, SARC, sarp.snowprofile, sarp.snowprofile.alignment, sarp.snowprofile.pyface, saseR, SC2API, scanMiR, scanMiRApp, scCustomize, SCdeconR, scDiffCom, scifer, scMET, scMitoMut, SCnorm, scoper, scopr, scorecard, ScoreEB, scoringutils, scPipe, scraEP, scRNAseqApp, scruff, scTreeViz, sdcHierarchies, sdcLog, sdcMicro, sdcTable, SDLfilter, SEA, seAMLess, SeaVal, seedr, segregation, seleniumPipes, SELF, semgram, SemNeT, semtree, sendigR, sense, sensobol, sentiment.ai, sentimentr, SenTinMixt, sentometrics, sentopics, seqsetvis, sequoia, SeqVarTools, servosphereR, SEtools, Seurat, sevenbridges, sevenbridges2, sevenC, sgat, SGP, SGPdata, SHAPforxgboost, shapr, shinyChatR, shinyepico, shinyExprPortal, shinyHugePlot, ShinyItemAnalysis, shinylogs, shinyML, ShinyQuickStarter, shinySbm, shinyWGD, siconvr, SightabilityModel, sigminer, Signac, signatureSearch, siie, SimBu, simcausal, simDAG, SimEngine, simfinapi, simitation, simPH, simphony, simplePHENOTYPES, simPop, simstudy, SimSurvey, simtrial, SingleCellSignalR, singleCellTK, sitepickR, siteymlgen, sitree, sitreeE, sjtable2df, skimr, skm, sleepr, SLEMI, SlideCNA, slim, SmallCountRounding, smallstuff, SmartEDA, smartsnp, smoothic, snapcount, SNPannotator, SNPhood, snplinkage, snpsettest, socialh, socialmixr, soilDB, SoilHyP, SoilTaxonomy, soiltestcorr, solaR2, solitude, SOMNiBUS, SOMnmR, somspace, soundgen, SpaceMarkers, spacyr, SpaDES.core, SpaDES.tools, sparrow, sparseFLMM, SPARSEMODr, SparseSignatures, spaths, SpatialCPie, SpatialFeatureExperiment, spatialHeatmap, SpatialOmicsOverlay, spatialrisk, SpATS, spatsoc, spDates, speaq, spectr, spectralAnalysis, spellcheckr, spicyR, SpliceWiz, splitstackshape, splutil, spNetwork, spocc, SPONGE, sport, sportyR, sqlHelpers, SQMtools, squallms, squid, SSBtools, SSHAARP, staggered, standartox, staRdom, statar, statcanR, statgenGWAS, statgenIBD, statgenMPP, statgenQTLxT, Statial, Statsomat, stdReg, stdReg2, stenR, sticky, SticsRFiles, stm, stochvolTMB, stplanr, stppSim, stringmagic, strvalidator, subscreen, subSeq, sumFREGAT, SUMMER, suncalc, SUNGEO, SunsVoc, superml, surveyplanning, surveyPrev, surveysd, survMisc, susographql, SVMDO, SVN, SWATH2stats, sweater, SWMPr, symphony, synlet, synr, table.express, tablecompare, tabledown, tableMatrix, tabulator, tabxplor, takos, TangledFeatures, targeted, targets, taskscheduleR, tatoo, taxize, taxonomizr, TCA, TCGAbiolinks, tcgaViz, tcpl, tcplfit2, tdsc, TeXCheckR, text2vec, textclean, TextMiningGUI, textplot, textpress, textrank, textrecipes, textshape, textTinyR, textTools, TFutils, TheOpenAIR, TideCurves, TideTables, tidyDenovix, TidyDensity, tidyfast, tidyfst, tidyft, tidygapminder, tidyplate, tidyr, tidyrules, tidysdm, tidytable, tidyterra, tidytext, tidytransit, tidyvpc, tiledb, timeplyr, timeseriesdb, TIN, tinytable, TitanCNA, tLOH, tmap, TMSig, TnT, toolStability, topdowntimeratio, tosca, totalcensus, ToxicoGx, toxSummary, TPAC, TPACData, tpm, TPP, tracee, trackdf, tradestatistics, TrafficBDE, trainsplit, transcriptogramer, transport, traveltimeR, treediff, treekoR, treemap, TreeMineR, treeshap, triact, TrialEmulation, tRigon, trimmer, tripr, TrumpetPlots, tsbox, tsdb, tsdistributions, tsgarch, tsmarch, tsmethods, TSrepr, TSstudio, tstests, tstools, TTAinterfaceTrendAnalysis, ttgsea, Tushare, tvtools, twang, twangRDC, twc, twl, twopartm, txshift, uavRmp, ubair, UCell, uci, udpipe, ugatsdb, UKB.COVID19, ukbnmr, ukbtools, UKgrid, Ularcirc, Unico, unifed, Uniquorn, uptasticsearch, utilsIPEA, vaersNDvax, vaersvax, validata, VanillaICE, vardpoor, vein, versioning, versus, VIM, vimp, ViSEAGO, visualpred, vivainsights, VMDecomp, VOSONDash, vosonSML, vote, votesys, VoxR, Voyager, vtreat, vvtableau, walkboutr, wally, WARDEN, weatherOz, WeatherSentiment, WebAnalytics, webtrackR, wehoop, WGScan, WhatsR, where, whSample, wikitaxa, wilson, wmm, word.alignment, WorldFlora, worrms, wpa, wpp2017, wpp2019, WQM, wrMisc, wrProteo, wTO, WVPlots, wxgenR, wyz.code.metaTesting, wyz.code.offensiveProgramming, wyz.code.rdoc, wyz.code.testthat, xcore, xefun, xesreadR, xgb2sql, xgboost, XNAString, xpectr, xplorerr, yamss, YEAB, youngSwimmers, zebu, zeitgebr, ZIM4rv, ZIprop, zoltr, zoomGroupStats</p>
</section>
<section id="namespace" class="level2">
<h2 class="anchored" data-anchor-id="namespace">Namespace</h2>
<p>Overall the package has 120 exported objects. 112 are documented:</p>
<pre><code>                :=                .BY             .EACHI               .GRP 
              TRUE               TRUE               TRUE               TRUE 
                .I      .Last.updated                 .N              .NGRP 
              TRUE               TRUE               TRUE               TRUE 
               .SD          %between%             %chin%            %flike% 
              TRUE              FALSE              FALSE              FALSE 
           %ilike%          %inrange%             %like%            %notin% 
             FALSE              FALSE              FALSE              FALSE 
           %plike%            address          alloc.col      as.data.table 
             FALSE               TRUE               TRUE               TRUE 
          as.IDate           as.ITime  as.xts.data.table            between 
              TRUE               TRUE               TRUE               TRUE 
           chgroup            chmatch            chorder                 CJ 
              TRUE               TRUE               TRUE               TRUE 
              copy               cube         data.table              dcast 
              TRUE               TRUE               TRUE               TRUE 
  dcast.data.table              fcase          fcoalesce        fdroplevels 
              TRUE               TRUE               TRUE               TRUE 
           fifelse         fintersect              first         format_col 
              TRUE               TRUE               TRUE               TRUE 
  format_list_item          foverlaps              frank             frankv 
              TRUE               TRUE               TRUE               TRUE 
             fread         frollapply          frollmean           frollsum 
              TRUE               TRUE               TRUE               TRUE 
          fsetdiff          fsetequal              fsort             funion 
              TRUE               TRUE               TRUE               TRUE 
            fwrite       getDTthreads getNumericRounding       groupingsets 
              TRUE               TRUE               TRUE               TRUE 
            haskey               hour          IDateTime            indices 
              TRUE               TRUE               TRUE               TRUE 
           inrange      is.data.table            isoweek                key 
              TRUE               TRUE               TRUE               TRUE 
             key&lt;-               last                let               like 
              TRUE               TRUE               TRUE               TRUE 
              mday            measure               melt    melt.data.table 
              TRUE               TRUE               TRUE               TRUE 
  merge.data.table             minute              month             nafill 
              TRUE               TRUE               TRUE               TRUE 
          patterns            quarter          rbindlist              rleid 
              TRUE               TRUE               TRUE               TRUE 
            rleidv             rollup              rowid             rowidv 
              TRUE               TRUE               TRUE               TRUE 
            second                set        setalloccol            setattr 
              TRUE               TRUE               TRUE               TRUE 
       setcolorder              setDF      setdroplevels              setDT 
              TRUE               TRUE               TRUE               TRUE 
      setDTthreads           setindex          setindexv             setkey 
              TRUE               TRUE               TRUE               TRUE 
           setkeyv          setnafill           setnames setNumericRounding 
              TRUE               TRUE               TRUE               TRUE 
          setorder          setorderv              shift        shouldPrint 
              TRUE               TRUE               TRUE               TRUE 
                SJ        substitute2             tables    test.data.table 
              TRUE               TRUE               TRUE               TRUE 
         timetaken          transpose         truelength          tstrsplit 
              TRUE               TRUE               TRUE               TRUE 
           uniqueN     update_dev_pkg               wday               week 
              TRUE               TRUE               TRUE               TRUE 
              yday               year            yearmon            yearqtr 
              TRUE               TRUE               TRUE               TRUE </code></pre>
<section id="examples" class="level3">
<h3 class="anchored" data-anchor-id="examples">Examples</h3>
<p>There are 67 help pages with examples, from 67 (100.00 %).</p>
</section>
<section id="news" class="level3">
<h3 class="anchored" data-anchor-id="news">NEWS</h3>
<p>The package has NEWS file and it is current.</p>
</section>
<section id="license" class="level3">
<h3 class="anchored" data-anchor-id="license">License</h3>
<p>The package uses .</p>
</section>
</section>
</section>
<section id="installation-environment" class="level1">
<h1>Installation environment</h1>
<section id="system-info" class="level2">
<h2 class="anchored" data-anchor-id="system-info">System Info</h2>
<table class="caption-top table">
<thead>
<tr class="header">
<th style="text-align: left;">Field</th>
<th style="text-align: left;">Value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Image</td>
<td style="text-align: left;">rhub/ref-image</td>
</tr>
<tr class="even">
<td style="text-align: left;">OS</td>
<td style="text-align: left;">Ubuntu 24.04.1 LTS</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Platform</td>
<td style="text-align: left;">x86_64-pc-linux-gnu</td>
</tr>
<tr class="even">
<td style="text-align: left;">System</td>
<td style="text-align: left;">x86_64, linux-gnu</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Execution Time</td>
<td style="text-align: left;">2025-02-17 04:55:41 UTC</td>
</tr>
</tbody>
</table>
<p><strong>System information</strong>. Table about the system used to check the package.</p>
</section>
<section id="r-session-info" class="level2">
<h2 class="anchored" data-anchor-id="r-session-info">R Session Info</h2>
<p>Information about the R environment and capabilities:</p>
<pre><code>R version 4.4.2 (2024-10-31)
Platform: x86_64-pc-linux-gnu
Running under: Ubuntu 24.04.1 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0

locale:
 [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       
 [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   
 [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          
[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   

time zone: UTC
tzcode source: system (glibc)

attached base packages:
[1] tools     stats     graphics  grDevices utils     datasets  methods  
[8] base     

loaded via a namespace (and not attached):
 [1] riskreports_0.0.0.9003 compiler_4.4.2         fastmap_1.2.0         
 [4] cli_3.6.4              htmltools_0.5.8.1      yaml_2.3.10           
 [7] rmarkdown_2.29         knitr_1.49             jsonlite_1.8.9        
[10] xfun_0.50              digest_0.6.37          rlang_1.1.5           
[13] evaluate_1.0.3        </code></pre>
<p>Platform used:</p>
<pre><code>   OS.type   file.sep dynlib.ext        GUI     endian    pkgType   path.sep 
    "unix"        "/"      ".so"      "X11"   "little"   "source"        ":" 
    r_arch 
        "" </code></pre>
<p>Râ€™s capabilities:</p>
<pre><code>       jpeg         png        tiff       tcltk         X11        aqua 
       TRUE        TRUE        TRUE        TRUE       FALSE       FALSE 
   http/ftp     sockets      libxml        fifo      cledit       iconv 
       TRUE        TRUE       FALSE        TRUE       FALSE        TRUE 
        NLS       Rprof     profmem       cairo         ICU long.double 
       TRUE        TRUE        TRUE        TRUE        TRUE        TRUE 
    libcurl 
       TRUE </code></pre>
<p>External software:</p>
<pre><code>                                                     zlib 
                                                    "1.3" 
                                                    bzlib 
                                     "1.0.8, 13-Jul-2019" 
                                                       xz 
                                                  "5.4.5" 
                                               libdeflate 
                                                   "1.19" 
                                                     PCRE 
                                       "10.42 2022-12-11" 
                                                      ICU 
                                                   "74.2" 
                                                      TRE 
                                "TRE 0.8.0 R_fixes (BSD)" 
                                                    iconv 
                                             "glibc 2.39" 
                                                 readline 
                                                    "8.2" 
                                                     BLAS 
"/usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3" </code></pre>
<p>Graphics external software:</p>
<pre><code>                   cairo                  cairoFT                    pango 
                "1.18.0"                       ""                 "1.52.1" 
                  libpng                     jpeg                  libtiff 
                "1.6.43"                    "8.0" "LIBTIFF, Version 4.5.1" </code></pre>
<p>Numerical characteristics of the machine:</p>
<pre><code>               double.eps            double.neg.eps               double.xmin 
             2.220446e-16              1.110223e-16             2.225074e-308 
              double.xmax               double.base             double.digits 
            1.797693e+308              2.000000e+00              5.300000e+01 
          double.rounding              double.guard         double.ulp.digits 
             5.000000e+00              0.000000e+00             -5.200000e+01 
    double.neg.ulp.digits           double.exponent            double.min.exp 
            -5.300000e+01              1.100000e+01             -1.022000e+03 
           double.max.exp               integer.max               sizeof.long 
             1.024000e+03              2.147484e+09              8.000000e+00 
          sizeof.longlong         sizeof.longdouble            sizeof.pointer 
             8.000000e+00              1.600000e+01              8.000000e+00 
            sizeof.time_t            longdouble.eps        longdouble.neg.eps 
             8.000000e+00              1.084202e-19              5.421011e-20 
        longdouble.digits       longdouble.rounding          longdouble.guard 
             6.400000e+01              5.000000e+00              0.000000e+00 
    longdouble.ulp.digits longdouble.neg.ulp.digits       longdouble.exponent 
            -6.300000e+01             -6.400000e+01              1.500000e+01 
       longdouble.min.exp        longdouble.max.exp 
            -1.638200e+04              1.638400e+04 </code></pre>
<p>Random number generation process:</p>
<pre><code>[1] "Mersenne-Twister" "Inversion"        "Rejection"       </code></pre>
</section>
<section id="information-about-the-environment" class="level2">
<h2 class="anchored" data-anchor-id="information-about-the-environment">Information about the environment</h2>
<p>Environmental and options variables affect how package checks and software in it might behave. Here is the environmental variables when running this report</p>
<pre><code>_R_CHECK_SYSTEM_CLOCK_
                        FALSE
ACCEPT_EULA             Y
ACTIONS_RUNNER_ACTION_ARCHIVE_CACHE
                        /opt/actionarchivecache
AGENT_TOOLSDIRECTORY    /opt/hostedtoolcache
ANDROID_HOME            /usr/local/lib/android/sdk
ANDROID_NDK             /usr/local/lib/android/sdk/ndk/27.2.12479018
ANDROID_NDK_HOME        /usr/local/lib/android/sdk/ndk/27.2.12479018
ANDROID_NDK_LATEST_HOME
                        /usr/local/lib/android/sdk/ndk/27.2.12479018
ANDROID_NDK_ROOT        /usr/local/lib/android/sdk/ndk/27.2.12479018
ANDROID_SDK_ROOT        /usr/local/lib/android/sdk
ANT_HOME                /usr/share/ant
AZURE_EXTENSION_DIR     /opt/az/azcliextensions
BOOTSTRAP_HASKELL_NONINTERACTIVE
                        1
BUNDLE_EXT              linux-amd64.deb
CHROME_BIN              /usr/bin/google-chrome
CHROMEWEBDRIVER         /usr/local/share/chromedriver-linux64
CI                      true
CONDA                   /usr/share/miniconda
DEBIAN_FRONTEND         noninteractive
DENO_DOM_PLUGIN         /opt/quarto/bin/tools/x86_64/deno_dom/libplugin.so
DENO_NO_UPDATE_CHECK    1
DENO_TLS_CA_STORE       system,mozilla
DEPLOYMENT_BASEPATH     /opt/runner
DOTNET_MULTILEVEL_LOOKUP
                        0
DOTNET_NOLOGO           1
DOTNET_SKIP_FIRST_TIME_EXPERIENCE
                        1
EDGEWEBDRIVER           /usr/local/share/edge_driver
EDITOR                  vi
GECKOWEBDRIVER          /usr/local/share/gecko_driver
GHCUP_INSTALL_BASE_PREFIX
                        /usr/local
GITHUB_ACTION           __run
GITHUB_ACTION_REF       
GITHUB_ACTION_REPOSITORY
                        
GITHUB_ACTIONS          true
GITHUB_ACTOR            llrs-roche
GITHUB_ACTOR_ID         185338939
GITHUB_API_URL          https://api.github.com
GITHUB_BASE_REF         
GITHUB_ENV              /home/runner/work/_temp/_runner_file_commands/set_env_24251222-dc77-41e6-994a-03413e3e9f5f
GITHUB_EVENT_NAME       schedule
GITHUB_EVENT_PATH       /home/runner/work/_temp/_github_workflow/event.json
GITHUB_GRAPHQL_URL      https://api.github.com/graphql
GITHUB_HEAD_REF         
GITHUB_JOB              main
GITHUB_OUTPUT           /home/runner/work/_temp/_runner_file_commands/set_output_24251222-dc77-41e6-994a-03413e3e9f5f
GITHUB_PATH             /home/runner/work/_temp/_runner_file_commands/add_path_24251222-dc77-41e6-994a-03413e3e9f5f
GITHUB_REF              refs/heads/main
GITHUB_REF_NAME         main
GITHUB_REF_PROTECTED    false
GITHUB_REF_TYPE         branch
GITHUB_REPOSITORY       pharmaR/pharmapkgs
GITHUB_REPOSITORY_ID    798326913
GITHUB_REPOSITORY_OWNER
                        pharmaR
GITHUB_REPOSITORY_OWNER_ID
                        42115094
GITHUB_RETENTION_DAYS   90
GITHUB_RUN_ATTEMPT      1
GITHUB_RUN_ID           13362121460
GITHUB_RUN_NUMBER       94
GITHUB_SERVER_URL       https://github.com
GITHUB_SHA              b25eba80408a4e3db53f5896d441ff2685638e62
GITHUB_STATE            /home/runner/work/_temp/_runner_file_commands/save_state_24251222-dc77-41e6-994a-03413e3e9f5f
GITHUB_STEP_SUMMARY     /home/runner/work/_temp/_runner_file_commands/step_summary_24251222-dc77-41e6-994a-03413e3e9f5f
GITHUB_TRIGGERING_ACTOR
                        llrs-roche
GITHUB_WORKFLOW         Update package repositories
GITHUB_WORKFLOW_REF     pharmaR/pharmapkgs/.github/workflows/update-repos.yml@refs/heads/main
GITHUB_WORKFLOW_SHA     b25eba80408a4e3db53f5896d441ff2685638e62
GITHUB_WORKSPACE        /home/runner/work/pharmapkgs/pharmapkgs
GOROOT_1_21_X64         /opt/hostedtoolcache/go/1.21.13/x64
GOROOT_1_22_X64         /opt/hostedtoolcache/go/1.22.12/x64
GOROOT_1_23_X64         /opt/hostedtoolcache/go/1.23.6/x64
GRADLE_HOME             /usr/share/gradle-8.12.1
HOME                    /home/runner
HOMEBREW_CLEANUP_PERIODIC_FULL_DAYS
                        3650
HOMEBREW_NO_AUTO_UPDATE
                        1
ImageOS                 ubuntu24
ImageVersion            20250209.1.0
INVOCATION_ID           d5e471a50fd941d195319ae0e3c382bb
JAVA_HOME               /usr/lib/jvm/temurin-17-jdk-amd64
JAVA_HOME_11_X64        /usr/lib/jvm/temurin-11-jdk-amd64
JAVA_HOME_17_X64        /usr/lib/jvm/temurin-17-jdk-amd64
JAVA_HOME_21_X64        /usr/lib/jvm/temurin-21-jdk-amd64
JAVA_HOME_8_X64         /usr/lib/jvm/temurin-8-jdk-amd64
JOURNAL_STREAM          8:6772
LANG                    C.UTF-8
LD_LIBRARY_PATH         /opt/R/4.4.2/lib/R/lib:/usr/local/lib:/usr/lib/x86_64-linux-gnu:/usr/lib/jvm/temurin-17-jdk-amd64/lib/server:/opt/R/4.4.2/lib/R/lib:/usr/local/lib:/usr/lib/x86_64-linux-gnu:/usr/lib/jvm/temurin-17-jdk-amd64/lib/server
LN_S                    ln -s
LOGGER_LOG_LEVEL        DEBUG
MAKE                    make
MEMORY_PRESSURE_WATCH   /sys/fs/cgroup/system.slice/runner-provisioner.service/memory.pressure
MEMORY_PRESSURE_WRITE   c29tZSAyMDAwMDAgMjAwMDAwMAA=
NOT_CRAN                true
NVM_DIR                 /home/runner/.nvm
PAGER                   /usr/bin/pager
PATH                    /snap/bin:/home/runner/.local/bin:/opt/pipx_bin:/home/runner/.cargo/bin:/home/runner/.config/composer/vendor/bin:/usr/local/.ghcup/bin:/home/runner/.dotnet/tools:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
PERFLOG_LOCATION_SETTING
                        RUNNER_PERFLOG
PHARMAPKGS_EXCLUDED_METRICS
                        assess_covr_coverage,assess_r_cmd_check
PHARMAPKGS_LIMIT        1000
PHARMAPKGS_PLATFORM     ubuntu-22.04
PHARMAPKGS_R_VERSION    4.4
PHARMAPKGS_REMOTE_REPO
                        https://cloud.r-project.org/
PIPX_BIN_DIR            /opt/pipx_bin
PIPX_HOME               /opt/pipx
PKGCACHE_HTTP_VERSION   2
PKGLOAD_PARENT_TEMPDIR
                        /tmp/RtmpjxUEUq
POWERSHELL_DISTRIBUTION_CHANNEL
                        GitHub-Actions-ubuntu24
PROCESSX_PS1bd6333a6894_1739768135
                        YES
PWD                     /home/runner/work/pharmapkgs/pharmapkgs
QUARTO_BIN_PATH         /opt/quarto/bin
QUARTO_DENO             /opt/quarto/bin/tools/x86_64/deno
QUARTO_DOCUMENT_PATH    /home/runner/work/pharmapkgs/pharmapkgs/inst/report
QUARTO_PROFILE          
QUARTO_PROJECT_DIR      /home/runner/work/pharmapkgs/pharmapkgs
QUARTO_PROJECT_ROOT     /home/runner/work/pharmapkgs/pharmapkgs
QUARTO_ROOT             /
QUARTO_SHARE_PATH       /opt/quarto/share
R_ARCH                  
R_BROWSER               xdg-open
R_BZIPCMD               /usr/bin/bzip2
R_DOC_DIR               /opt/R/4.4.2/lib/R/doc
R_GZIPCMD               /usr/bin/gzip
R_HOME                  /opt/R/4.4.2/lib/R
R_INCLUDE_DIR           /opt/R/4.4.2/lib/R/include
R_LIB_FOR_PAK           /opt/R/4.4.2/lib/R/site-library
R_LIBS_SITE             /opt/R/4.4.2/lib/R/site-library
R_LIBS_USER             /home/runner/work/_temp/Library
R_PAPERSIZE             letter
R_PAPERSIZE_USER        letter
R_PDFVIEWER             /usr/bin/xdg-open
R_PLATFORM              x86_64-pc-linux-gnu
R_PRINTCMD              /usr/bin/lpr
R_RD4PDF                times,inconsolata,hyper
R_SESSION_TMPDIR        /tmp/RtmpngjTMs
R_SHARE_DIR             /opt/R/4.4.2/lib/R/share
R_STRIP_SHARED_LIB      strip --strip-unneeded
R_STRIP_STATIC_LIB      strip --strip-debug
R_TEXI2DVICMD           /usr/bin/texi2dvi
R_UNZIPCMD              /usr/bin/unzip
R_ZIPCMD                /usr/bin/zip
RUNNER_ARCH             X64
RUNNER_ENVIRONMENT      github-hosted
RUNNER_NAME             GitHub Actions 16
RUNNER_OS               Linux
RUNNER_PERFLOG          /home/runner/perflog
RUNNER_TEMP             /home/runner/work/_temp
RUNNER_TOOL_CACHE       /opt/hostedtoolcache
RUNNER_TRACKING_ID      github_84602b48-6461-4335-a438-3ca558c890e8
RUNNER_USER             runner
RUNNER_WORKSPACE        /home/runner/work/pharmapkgs
SED                     /usr/bin/sed
SELENIUM_JAR_PATH       /usr/share/java/selenium-server.jar
SGX_AESM_ADDR           1
SHLVL                   1
STATS_D                 true
STATS_D_D               true
STATS_D_TC              true
STATS_EXT               true
STATS_EXTP              https://provjobdprod.z13.web.core.windows.net/settings/provjobdsettings-latest/provjobd.data
STATS_PIP               false
STATS_RDCL              true
STATS_TRP               true
STATS_UE                true
STATS_V3PS              true
STATS_VMD               true
STATS_VMFE              true
SWIFT_PATH              /usr/share/swift/usr/bin
SYSTEMD_EXEC_PID        830
TAR                     /usr/bin/tar
TZ                      UTC
USER                    runner
VCPKG_INSTALLATION_ROOT
                        /usr/local/share/vcpkg
XDG_CONFIG_HOME         /home/runner/.config
XDG_RUNTIME_DIR         /run/user/1001</code></pre>
<p>These are the options set to generate the report:</p>
<pre><code>$add.smooth
[1] TRUE

$bitmapType
[1] "cairo"

$browser
[1] "xdg-open"

$browserNLdisabled
[1] FALSE

$callr.condition_handler_cli_message
function (msg) 
{
    custom_handler &lt;- getOption("cli.default_handler")
    if (is.function(custom_handler)) {
        custom_handler(msg)
    }
    else {
        cli_server_default(msg)
    }
}
&lt;bytecode: 0x55732fe64910&gt;
&lt;environment: namespace:cli&gt;

$catch.script.errors
[1] FALSE

$CBoundsCheck
[1] FALSE

$check.bounds
[1] FALSE

$citation.bibtex.max
[1] 1

$continue
[1] "+ "

$contrasts
        unordered           ordered 
"contr.treatment"      "contr.poly" 

$covr.record_tests
[1] TRUE

$defaultPackages
[1] "datasets"  "utils"     "grDevices" "graphics"  "stats"     "methods"  

$demo.ask
[1] "default"

$deparse.cutoff
[1] 60

$device
function (width = 7, height = 7, ...) 
{
    grDevices::pdf(NULL, width, height, ...)
}
&lt;bytecode: 0x55732f54cb50&gt;
&lt;environment: namespace:knitr&gt;

$device.ask.default
[1] FALSE

$digits
[1] 7

$dvipscmd
[1] "dvips"

$echo
[1] FALSE

$editor
[1] "vi"

$encoding
[1] "native.enc"

$example.ask
[1] "default"

$expressions
[1] 5000

$help.search.types
[1] "vignette" "demo"     "help"    

$help.try.all.packages
[1] FALSE

$htmltools.preserve.raw
[1] TRUE

$HTTPUserAgent
[1] "R/4.4.2 R (4.4.2 x86_64-pc-linux-gnu x86_64 linux-gnu) on GitHub Actions"

$internet.info
[1] 2

$keep.parse.data
[1] TRUE

$keep.parse.data.pkgs
[1] FALSE

$keep.source
[1] FALSE

$keep.source.pkgs
[1] FALSE

$knitr.in.progress
[1] TRUE

$locatorBell
[1] TRUE

$mailer
[1] "mailto"

$matprod
[1] "default"

$max.contour.segments
[1] 25000

$max.print
[1] 99999

$menu.graphics
[1] TRUE

$na.action
[1] "na.omit"

$Ncpus
[1] 1

$nwarnings
[1] 50

$OutDec
[1] "."

$pager
[1] "/opt/R/4.4.2/lib/R/bin/pager"

$papersize
[1] "letter"

$PCRE_limit_recursion
[1] NA

$PCRE_study
[1] FALSE

$PCRE_use_JIT
[1] TRUE

$pdfviewer
[1] "/usr/bin/xdg-open"

$pkgType
[1] "source"

$printcmd
[1] "/usr/bin/lpr"

$prompt
[1] "&gt; "

$repos
                      CRAN 
"https://cran.rstudio.com" 

$rl_word_breaks
[1] " \t\n\"\\'`&gt;&lt;=%;,|&amp;{()}"

$rlang_trace_top_env
&lt;environment: R_GlobalEnv&gt;

$scipen
[1] 0

$show.coef.Pvalues
[1] TRUE

$show.error.messages
[1] TRUE

$show.signif.stars
[1] TRUE

$showErrorCalls
[1] TRUE

$showNCalls
[1] 50

$showWarnCalls
[1] FALSE

$str
$str$strict.width
[1] "no"

$str$digits.d
[1] 3

$str$vec.len
[1] 4

$str$list.len
[1] 99

$str$deparse.lines
NULL

$str$drop.deparse.attr
[1] TRUE

$str$formatNum
function (x, ...) 
format(x, trim = TRUE, drop0trailing = TRUE, ...)
&lt;environment: 0x55732f33c520&gt;


$str.dendrogram.last
[1] "`"

$texi2dvi
[1] "/usr/bin/texi2dvi"

$tikzMetricsDictionary
[1] "validation_report_data.table_v1.16.4-tikzDictionary"

$timeout
[1] 60

$try.outFile
A connection with                    
description ""      
class       "file"  
mode        "w+b"   
text        "binary"
opened      "opened"
can read    "yes"   
can write   "yes"   

$ts.eps
[1] 1e-05

$ts.S.compat
[1] FALSE

$unzip
[1] "/usr/bin/unzip"

$useFancyQuotes
[1] FALSE

$verbose
[1] FALSE

$warn
[1] 0

$warning.length
[1] 1000

$warnPartialMatchArgs
[1] FALSE

$warnPartialMatchAttr
[1] FALSE

$warnPartialMatchDollar
[1] FALSE

$width
[1] 80</code></pre>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
    const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
    const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
    let newTheme = '';
    if(darkModeDefault) {
      newTheme = isAlternate ? baseTheme : alternateTheme;
    } else {
      newTheme = isAlternate ? alternateTheme : baseTheme;
    }
    const changeGiscusTheme = () => {
      // From: https://github.com/giscus/giscus/issues/336
      const sendMessage = (message) => {
        const iframe = document.querySelector('iframe.giscus-frame');
        if (!iframe) return;
        iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
      }
      sendMessage({
        setConfig: {
          theme: newTheme
        }
      });
    }
    const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
    if (isGiscussLoaded) {
      changeGiscusTheme();
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  const darkModeDefault = false;
  let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
    toggleGiscusIfUsed(toAlternate, darkModeDefault);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/pharmar\.github\.io\/pharmapkgs");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




<footer class="footer"><div class="nav-footer"><div class="nav-footer-center"><div class="toc-actions"><ul><li><a href="https://github.com/pharmaR/pharmapkgs/issues/new" class="toc-action"><i class="bi bi-github"></i>Report an issue</a></li></ul></div></div></div></footer></body></html>